<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>CODIS Marker Allele Frequencies</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="demo-data.html">Demographic Data</a>
</li>
<li>
  <a href="database-composition.html">Database composition</a>
</li>
<li>
  <a href="STR-simulation.html">STR simulations</a>
</li>
<li>
  <a href="CODIS-STR-freqs.html">CODIS STRs</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tinalasisi/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CODIS Marker Allele Frequencies</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2024-07-29 10:45:03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree873eb70f293309e6179fb98bc2f18143d1580153targetblank873eb70a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/873eb70f293309e6179fb98bc2f18143d1580153" target="_blank">873eb70</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree873eb70f293309e6179fb98bc2f18143d1580153targetblank873eb70a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/873eb70f293309e6179fb98bc2f18143d1580153" target="_blank">873eb70</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/CODIS-STR-freqs.Rmd</code>) and
HTML (<code>docs/CODIS-STR-freqs.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/fbdfcf1e0d6544306069114e6a1850d8311ef93d/analysis/CODIS-STR-freqs.Rmd" target="_blank">fbdfcf1</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-23
</td>
<td>
Fixed allele frequency table and continued simulation development
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/CODIS-STR-freqs.html" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/analysis/CODIS-STR-freqs.Rmd" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
wflow_rename("analysis/shortrange.Rmd", "analysis/CODIS-STR-freqs.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/CODIS-STR-freqs.html" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
wflow_rename("analysis/shortrange.Rmd", "analysis/CODIS-STR-freqs.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="codis-marker-allele-frequencies" class="section level2">
<h2>CODIS marker allele frequencies</h2>
<p>Frequencies and raw genotypes for different populations were found <a
href="https://www.promega.com/products/pm/genetic-identity/population-statistics/allele-frequencies/">here</a>
and refer to Steffen, C.R., Coble, M.D., Gettings, K.B., Vallone, P.M.
(2017) Corrigendum to ‘U.S. Population Data for 29 Autosomal STR Loci’
[Forensic Sci. Int. Genet. 7 (2013) e82-e83]. Forensic Sci. Int. Genet.
31, e36–e40. The US core CODIS markers are a subset of the 29 described
here.</p>
</div>
<div id="load-codis-allele-frequencies" class="section level2">
<h2>Load CODIS allele frequencies</h2>
<p>CODIS allele frequencies were found through NIST STR base and
specifically downloaded from the supplementary materials of Steffen et
al 2017. These are 1036 unrelated individuals from the U.S.
population.</p>
<pre class="r"><code>library(readxl)
library(tidyverse)
library(data.table)

# Define the file path
file_path &lt;- &quot;data/1036_allelefreqs.xlsx&quot;

# Function to map sheet names to population names
map_population_name &lt;- function(sheet_name) {
  case_when(
    str_detect(sheet_name, &quot;all&quot;) ~ &quot;all&quot;,
    str_detect(sheet_name, &quot;AfAm&quot;) ~ &quot;AfAm&quot;,
    str_detect(sheet_name, &quot;Cauc&quot;) ~ &quot;Cauc&quot;,
    str_detect(sheet_name, &quot;Hispanic&quot;) ~ &quot;Hispanic&quot;,
    str_detect(sheet_name, &quot;Asian&quot;) ~ &quot;Asian&quot;,
    TRUE ~ &quot;Unknown&quot;
  )
}

# Function to read each sheet and process the data
read_and_process_sheet &lt;- function(sheet) {
  df &lt;- read_excel(file_path, sheet = sheet, range = &quot;A5:AD101&quot;, col_names = FALSE)
  colnames(df) &lt;- c(&quot;allele&quot;, &quot;CSF1PO&quot;, &quot;D10S1248&quot;, &quot;D12S391&quot;, &quot;D13S317&quot;, &quot;D16S539&quot;, &quot;D18S51&quot;, &quot;D19S433&quot;, &quot;D1S1656&quot;, 
                    &quot;D21S11&quot;, &quot;D22S1045&quot;, &quot;D2S1338&quot;, &quot;D2S441&quot;, &quot;D3S1358&quot;, &quot;D5S818&quot;, &quot;D6S1043&quot;, &quot;D7S820&quot;, &quot;D8S1179&quot;, 
                    &quot;F13A01&quot;, &quot;F13B&quot;, &quot;FESFPS&quot;, &quot;FGA&quot;, &quot;LPL&quot;, &quot;Penta_C&quot;, &quot;Penta_D&quot;, &quot;Penta_E&quot;, &quot;SE33&quot;, &quot;TH01&quot;, &quot;TPOX&quot;, &quot;vWA&quot;)
  
  df &lt;- df %&gt;%
    mutate(allele = as.character(allele)) %&gt;%
    pivot_longer(-allele, names_to = &quot;marker&quot;, values_to = &quot;frequency&quot;) %&gt;%
    mutate(population = map_population_name(sheet)) %&gt;%
    replace_na(list(frequency = 0))
  
  return(df)
}

# Read all sheets into a list of dataframes and combine them
sheet_names &lt;- excel_sheets(file_path)
sheets_data &lt;- lapply(sheet_names, read_and_process_sheet)
df_allelefreq &lt;- bind_rows(sheets_data)

# Convert to data.table
setDT(df_allelefreq)

# Print the combined dataframe to check
print(head(df_allelefreq))</code></pre>
<pre><code>   allele   marker frequency population
   &lt;char&gt;   &lt;char&gt;     &lt;num&gt;     &lt;char&gt;
1:    2.2   CSF1PO         0        all
2:    2.2 D10S1248         0        all
3:    2.2  D12S391         0        all
4:    2.2  D13S317         0        all
5:    2.2  D16S539         0        all
6:    2.2   D18S51         0        all</code></pre>
<pre class="r"><code># Check if the population column is correct
print(unique(df_allelefreq$population))</code></pre>
<pre><code>[1] &quot;all&quot;      &quot;AfAm&quot;     &quot;Cauc&quot;     &quot;Hispanic&quot; &quot;Asian&quot;   </code></pre>
<pre class="r"><code># Save the combined dataframe to a single CSV file
fwrite(df_allelefreq, &quot;data/df_allelefreq_combined.csv&quot;)

# Function to sort alleles numerically while treating them as characters
sort_alleles &lt;- function(df) {
  df %&gt;% 
    mutate(allele = as.numeric(allele)) %&gt;%
    arrange(allele) %&gt;%
    mutate(allele = as.character(allele))
}

# Create and save a dataframe per population with alleles as the first column and markers as subsequent columns
populations &lt;- unique(df_allelefreq$population)
for (pop in populations) {
  df_pop &lt;- df_allelefreq[population == pop] %&gt;%
    pivot_wider(names_from = marker, values_from = frequency) %&gt;%
    sort_alleles()
  fwrite(df_pop, paste0(&quot;data/1036_&quot;, pop, &quot;_wide.csv&quot;))
}

# Create the wide-format dataframe for all populations
df_allelefreq_wide &lt;- df_allelefreq %&gt;%
  pivot_wider(names_from = population, values_from = frequency) %&gt;%
  replace_na(list(all = 0, AfAm = 0, Cauc = 0, Hispanic = 0, Asian = 0))

# Convert to data.table
setDT(df_allelefreq_wide)

# Sort the wide dataframe alleles numerically
df_allelefreq_wide &lt;- sort_alleles(df_allelefreq_wide)

# Print the wide dataframe to check
# print(head(df_allelefreq_wide))

# Save the wide-format dataframe to a CSV file
fwrite(df_allelefreq_wide, &quot;data/df_allelefreq_wide.csv&quot;)

# Convert allele to numeric in the df_allelefreq table
df &lt;- copy(df_allelefreq)
df[, allele := as.numeric(allele)]

# Print the modified dataframe to check
# print(head(df))
# 
# # Filter out rows where frequency is 0
df&lt;- df %&gt;%
  filter(frequency != 0)</code></pre>
<div id="box-plot-of-allele-frequencies" class="section level3">
<h3>Box plot of allele frequencies</h3>
<p>Below we show a box plot of the distribution of allele frequencies
for each marker, grouped by population.</p>
<pre class="r"><code>df %&gt;%
  group_by(population) %&gt;%
  filter(population != &quot;all&quot;) %&gt;%
  ggplot(aes(x = marker, y = allele, fill = population)) +
  geom_boxplot() +
  labs(x = &quot;Marker&quot;, y = &quot;Allele&quot;,
       title = &quot;Range of Alleles Across Different Markers (Grouped by Population)&quot;) +
  theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-boxplot-all-group-allelefreqs-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-boxplot-all-group-allelefreqs-1">
Past versions of plt-boxplot-all-group-allelefreqs-1.png
</button>
</p>
<div id="fig-plt-boxplot-all-group-allelefreqs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-boxplot-all-group-allelefreqs-1.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="histogram-of-allele-frequencies" class="section level3">
<h3>Histogram of allele frequencies</h3>
<p>Below we show a histogram of the distribution of allele frequencies
for each marker, grouped by population.</p>
<pre class="r"><code># Find unique markers and split them into chunks of 5
unique_markers &lt;- unique(df$marker)
marker_chunks &lt;- split(unique_markers, ceiling(seq_along(unique_markers)/5))

# Loop through the chunks and create a plot for each chunk
for(i in seq_along(marker_chunks)) {
  df_chunk &lt;- df %&gt;% filter(marker %in% marker_chunks[[i]])
  
  p &lt;- ggplot(df_chunk, aes(x = allele, y = frequency, fill = population)) +
    geom_col(position = &quot;dodge&quot;, width = 0.7) +
    facet_grid(population ~ marker) +
    # scale_x_continuous(breaks = seq(2.2, 43.2, by = 1)) +
    labs(x = &quot;Allele&quot;, y = &quot;Frequency&quot;,
         title = paste(&quot;Allele Frequencies for Markers&quot;, i*5-4, &quot;to&quot;, min(i*5, length(unique_markers))),
         caption = &quot;Source: CSV data&quot;) +
    theme_bw()

  print(p)
}</code></pre>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-1">
Past versions of plt-histo-all-group-allelefreqs-1.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-1.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-2">
Past versions of plt-histo-all-group-allelefreqs-2.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-2.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-3">
Past versions of plt-histo-all-group-allelefreqs-3.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-3.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-4">
Past versions of plt-histo-all-group-allelefreqs-4.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-4.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-5">
Past versions of plt-histo-all-group-allelefreqs-5.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-5.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-6.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-histo-all-group-allelefreqs-6">
Past versions of plt-histo-all-group-allelefreqs-6.png
</button>
</p>
<div id="fig-plt-histo-all-group-allelefreqs-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/plt-histo-all-group-allelefreqs-6.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="summary-of-allele-frequencies" class="section level3">
<h3>Summary of allele frequencies</h3>
<pre class="r"><code>summary_df &lt;- df %&gt;%
  group_by(marker, population) %&gt;%
  summarise(num_alleles = n_distinct(allele))</code></pre>
<pre class="r"><code>library(knitr)
library(kableExtra)

summary_df &lt;- df %&gt;%
  group_by(marker, population) %&gt;%
  summarise(num_alleles = n_distinct(allele), .groups = &#39;drop&#39;)

summary_pivot &lt;- summary_df %&gt;%
  pivot_wider(names_from = population, values_from = num_alleles)

# Print the table using kable and kableExtra for styling
summary_pivot %&gt;%
  kable(&quot;markdown&quot;, caption = &quot;Summary of Distinct Alleles by Marker and Population&quot;) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;, &quot;responsive&quot;))</code></pre>
<table class="table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<caption>
Summary of Distinct Alleles by Marker and Population
</caption>
<thead>
<tr>
<th style="text-align:left;">
marker
</th>
<th style="text-align:right;">
AfAm
</th>
<th style="text-align:right;">
Asian
</th>
<th style="text-align:right;">
Cauc
</th>
<th style="text-align:right;">
Hispanic
</th>
<th style="text-align:right;">
all
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CSF1PO
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
D10S1248
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
D12S391
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
D13S317
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
D16S539
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
D18S51
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
22
</td>
</tr>
<tr>
<td style="text-align:left;">
D19S433
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
D1S1656
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
D21S11
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
D22S1045
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
D2S1338
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
D2S441
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
D3S1358
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
D5S818
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
D6S1043
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
D7S820
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
D8S1179
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
F13A01
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
F13B
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
FESFPS
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
FGA
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
LPL
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
Penta_C
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
Penta_D
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
Penta_E
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
23
</td>
</tr>
<tr>
<td style="text-align:left;">
SE33
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
TH01
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
TPOX
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
vWA
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="simulating-genotypes" class="section level2">
<h2>Simulating genotypes</h2>
<div id="one-marker-at-a-time" class="section level3">
<h3>One marker at a time</h3>
<pre class="r"><code># Function to simulate genotypes for a pair of individuals
simulate_genotypes &lt;- function(allele_frequencies, relationship_type) {
  # Draw two alleles for the first individual from the population allele frequencies
  individual1 &lt;- sample(names(allele_frequencies), size = 2, replace = TRUE, prob = allele_frequencies)
  
  # Define the probabilities of sharing 0, 1, or 2 IBD alleles for each relationship type
  relationship_probs &lt;- list(
    &#39;parent_child&#39; = c(0, 1, 0), # In a parent-child relationship, always 1 allele is shared
    &#39;full_siblings&#39; = c(1/4, 1/2, 1/4), # For full siblings, the probabilities are 1/4 for sharing 0, 1/2 for sharing 1, and 1/4 for sharing 2 alleles
    &#39;half_siblings&#39; = c(1/2, 1/2, 0), # For half siblings, the probabilities are 1/2 for sharing 0 and 1/2 for sharing 1 allele
    &#39;cousins&#39; = c(7/8, 1/8, 0), # For cousins, the probabilities are 7/8 for sharing 0 and 1/8 for sharing 1 allele
    &#39;second_cousins&#39; = c(15/16, 1/16, 0), # For second cousins, the probabilities are 15/16 for sharing 0 and 1/16 for sharing 1 allele
    &#39;unrelated&#39; = c(1, 0, 0) # For unrelated individuals, they always share 0 alleles
  )
  
  # Get the probabilities of sharing alleles for the specific relationship type
  prob_shared_alleles &lt;- relationship_probs[[relationship_type]]
  
  # Draw the number of shared alleles based on these probabilities
  num_shared_alleles &lt;- sample(c(0, 1, 2), size = 1, prob = prob_shared_alleles)
  
  # Construct the genotype of the second individual by sampling the shared alleles from the first individual 
  # and the rest from the population allele frequencies
  individual2 &lt;- c(sample(individual1, size = num_shared_alleles), sample(names(allele_frequencies), size = 2 - num_shared_alleles, replace = TRUE, prob = allele_frequencies))
  
  # Return the genotypes of the two individuals and the number of shared alleles
  return(list(individual1 = individual1, individual2 = individual2, num_shared_alleles = num_shared_alleles))
}

# Function to calculate the index of relatedness
calculate_relatedness &lt;- function(simulated_genotypes, allele_frequencies) {
  # Calculate the number of alleles that the two individuals share
  num_shared_alleles &lt;- sum(simulated_genotypes$individual1 %in% simulated_genotypes$individual2)
  
  # Calculate the index of relatedness as the number of shared alleles divided by the sum of the inverse of allele frequencies 
  # of the alleles in the genotypes of both individuals. This gives a higher weight to rare alleles.
  relatedness &lt;- num_shared_alleles / (sum(1 / allele_frequencies[simulated_genotypes$individual1]) + sum(1 / allele_frequencies[simulated_genotypes$individual2]))
  
    # Return the index of relatedness and the number of shared alleles
  return(list(relatedness = relatedness, num_shared_alleles = simulated_genotypes$num_shared_alleles))
}

# Function to simulate genotypes and calculate relatedness for different relationships
simulate_relatedness &lt;- function(df, marker, population, relationship_type) {
  # Filter the data frame to get the allele frequencies for the specific marker and population
  allele_frequencies &lt;- df %&gt;%
    filter(marker == marker, population == population) %&gt;%
    pull(frequency) %&gt;%
    setNames(df$allele)
  
  # Simulate the genotypes for the pair of individuals using these allele frequencies and the specific relationship type
  simulated_genotypes &lt;- simulate_genotypes(allele_frequencies, relationship_type)
  
  # Calculate the relatedness index based on these simulated genotypes and the allele frequencies
  relatedness_data &lt;- calculate_relatedness(simulated_genotypes, allele_frequencies)
  
  # Return the marker, population, relationship type, and the calculated relatedness data
  return(c(list(marker = marker, population = population, relationship_type = relationship_type), relatedness_data))
}

# Example usage
simulate_relatedness(df, marker = &quot;F13A01&quot;, population = &quot;Asian&quot;, relationship_type = &quot;full_siblings&quot;)</code></pre>
<pre><code>$marker
[1] &quot;F13A01&quot;

$population
[1] &quot;Asian&quot;

$relationship_type
[1] &quot;full_siblings&quot;

$relatedness
[1] 0

$num_shared_alleles
[1] 0</code></pre>
<pre class="r"><code>simulate_all_relationships &lt;- function(df, num_simulations) {
  # Define the list of relationship types
  relationship_types &lt;- c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)
  
  # Initialize an empty list to store results
  results &lt;- list()
  
  # Iterate over all combinations of markers, populations, and relationship types
  for (marker in unique(df$marker)) {
    for (population in unique(df$population)) {
      for (relationship_type in relationship_types) {
        for (i in 1:num_simulations) {
          # Simulate relatedness and add the result to the list
          results[[length(results) + 1]] &lt;- simulate_relatedness(df, marker, population, relationship_type)
        }
      }
    }
  }
  
  # Convert the list of results to a dataframe
  results_df &lt;- do.call(rbind, lapply(results, function(x) as.data.frame(t(unlist(x)))))
  
  return(results_df)
}</code></pre>
<pre class="r"><code># Usage:
# df &lt;- # Your dataframe here
results_df &lt;- simulate_all_relationships(df, num_simulations = 10)</code></pre>
<div id="visualization" class="section level4">
<h4>Visualization</h4>
<pre class="r"><code># Function to capitalize the first letter of a string
ucfirst &lt;- function(s) {
  paste(toupper(substring(s, 1,1)), substring(s, 2), sep = &quot;&quot;)
}

create_plot &lt;- function(df, variable_to_plot) {
  # Create the plot
  p &lt;- ggplot(df, aes(x = relationship_type, y = .data[[variable_to_plot]], color = population, shape = population, fill = population)) +
    geom_boxplot(alpha = 0.4) + # Change the order of geom_boxplot() and geom_point() and adjust alpha
    geom_point(position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75), size = 1, alpha = 0.6) +
    facet_grid(. ~ marker) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    scale_x_discrete(limits = c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)) +
    labs(title = paste(ucfirst(variable_to_plot), &quot;by Relationship Type, Population, and STR Marker&quot;),
         x = &quot;Relationship Type&quot;, 
         y = ucfirst(variable_to_plot), 
         color = &quot;Population&quot;,
         shape = &quot;Population&quot;)
  
  return(p)
}</code></pre>
<pre class="r"><code># Filter your data for the first 3 unique STR markers
factor_vars &lt;- c(&quot;marker&quot;, &quot;population&quot;, &quot;relationship_type&quot;)
numeric_vars &lt;- c(&quot;relatedness&quot;, &quot;num_shared_alleles&quot;)

# Filter your data for unique STR markers and remove the &quot;all&quot; population
results_df_filtered &lt;- results_df %&gt;%
  filter(population != &quot;all&quot;) %&gt;%
  mutate(
    across(all_of(factor_vars), as.factor),
    across(all_of(numeric_vars), as.numeric)
  )

# Find the total number of unique markers
total_markers &lt;- length(unique(results_df_filtered$marker))

# Iterate through the unique markers in groups of 3
for (i in seq(1, total_markers, by = 3)) {
  # Filter the data for the current set of 3 or fewer markers
  filtered_results_df &lt;- results_df_filtered %&gt;%
    filter(marker %in% unique(marker)[i:min(i+2, total_markers)])
  
  # Create and display the plots for the current set of markers
  for (variable_to_plot in c(&quot;relatedness&quot;, &quot;num_shared_alleles&quot;)) {
    plot &lt;- create_plot(df = filtered_results_df, variable_to_plot = variable_to_plot)
    print(plot) # Print the plot to display it in the RMarkdown document
  }
}</code></pre>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-1">
Past versions of plt-single-1.png
</button>
</p>
<div id="fig-plt-single-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-1.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-2">
Past versions of plt-single-2.png
</button>
</p>
<div id="fig-plt-single-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-2.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-3">
Past versions of plt-single-3.png
</button>
</p>
<div id="fig-plt-single-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-3.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-4">
Past versions of plt-single-4.png
</button>
</p>
<div id="fig-plt-single-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-4.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-5">
Past versions of plt-single-5.png
</button>
</p>
<div id="fig-plt-single-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-5.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-6">
Past versions of plt-single-6.png
</button>
</p>
<div id="fig-plt-single-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-6.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-7">
Past versions of plt-single-7.png
</button>
</p>
<div id="fig-plt-single-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-7.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-8">
Past versions of plt-single-8.png
</button>
</p>
<div id="fig-plt-single-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-8.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-9">
Past versions of plt-single-9.png
</button>
</p>
<div id="fig-plt-single-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-9.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-10">
Past versions of plt-single-10.png
</button>
</p>
<div id="fig-plt-single-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-10.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-11">
Past versions of plt-single-11.png
</button>
</p>
<div id="fig-plt-single-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-11.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-12">
Past versions of plt-single-12.png
</button>
</p>
<div id="fig-plt-single-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-12.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-13">
Past versions of plt-single-13.png
</button>
</p>
<div id="fig-plt-single-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-13.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-14">
Past versions of plt-single-14.png
</button>
</p>
<div id="fig-plt-single-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-14.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-15">
Past versions of plt-single-15.png
</button>
</p>
<div id="fig-plt-single-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-15.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-16">
Past versions of plt-single-16.png
</button>
</p>
<div id="fig-plt-single-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-16.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-17">
Past versions of plt-single-17.png
</button>
</p>
<div id="fig-plt-single-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-17.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-18.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-18">
Past versions of plt-single-18.png
</button>
</p>
<div id="fig-plt-single-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-18.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-19.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-19">
Past versions of plt-single-19.png
</button>
</p>
<div id="fig-plt-single-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-19.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/CODIS-STR-freqs.Rmd/plt-single-20.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-single-20">
Past versions of plt-single-20.png
</button>
</p>
<div id="fig-plt-single-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/da5eacd910c4d9bd7134018cea7d681479b5a98e/docs/figure/CODIS-STR-freqs.Rmd/plt-single-20.png" target="_blank">da5eacd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="all-markers-together" class="section level3">
<h3>All Markers together</h3>
<pre class="r"><code># Function to simulate genotypes for a pair of individuals for all markers
simulate_genotypes_all_markers &lt;- function(df, relationship_type, population) {
  markers &lt;- unique(df$marker)
  
  # Simulate the first individual&#39;s alleles by drawing from the population frequency for each marker
  individual1 &lt;- setNames(lapply(markers, function(marker) {
    allele_frequencies &lt;- df %&gt;%
      filter(marker == marker, population == population) %&gt;%
      pull(frequency) %&gt;%
      setNames(df$allele)
      
    sample(names(allele_frequencies), size = 2, replace = TRUE, prob = allele_frequencies)
  }), markers)
  
  # Relationship probabilities
  relationship_probs &lt;- list(
    &#39;parent_child&#39; = c(0, 1, 0),
    &#39;full_siblings&#39; = c(1/4, 1/2, 1/4),
    &#39;half_siblings&#39; = c(1/2, 1/2, 0),
    &#39;cousins&#39; = c(7/8, 1/8, 0),
    &#39;second_cousins&#39; = c(15/16, 1/16, 0),
    &#39;unrelated&#39; = c(1, 0, 0)
  )
  
  prob_shared_alleles &lt;- relationship_probs[[relationship_type]]
  num_shared_alleles &lt;- sample(c(0, 1, 2), size = 1, prob = prob_shared_alleles)
  
  individual2 &lt;- setNames(lapply(markers, function(marker) {
  allele_frequencies &lt;- df %&gt;%
    filter(marker == marker, population == population) %&gt;%
    arrange(marker, allele) %&gt;%
    pull(frequency) %&gt;%
    setNames(df$allele)
  
  prob_shared_alleles &lt;- relationship_probs[[relationship_type]]
  non_zero_indices &lt;- which(prob_shared_alleles != 0)
  num_shared_alleles &lt;- sample(non_zero_indices - 1, size = 1, prob = prob_shared_alleles[non_zero_indices])

  alleles_from_individual1 &lt;- sample(individual1[[marker]], size = num_shared_alleles)
  alleles_from_population &lt;- sample(names(allele_frequencies), size = 2 - num_shared_alleles, replace = TRUE, prob = allele_frequencies)
  
  return(c(alleles_from_individual1, alleles_from_population))
}), markers)

  # Return the simulated genotypes
  return(list(individual1 = individual1, individual2 = individual2))
}



# Function to calculate the index of relatedness for all markers
calculate_relatedness_all_markers &lt;- function(simulated_genotypes, df, population) {
  markers &lt;- names(simulated_genotypes$individual1)
  
  # Calculate the number of shared alleles for each marker
  num_shared_alleles &lt;- sapply(markers, function(marker) {
    sum(simulated_genotypes$individual1[[marker]] %in% simulated_genotypes$individual2[[marker]])
  })
  
  # Calculate the index of relatedness as the number of shared alleles weighted inversely to their frequencies
  # Now considering both individuals&#39; alleles for the inverse frequency weighting
  relatedness &lt;- sapply(markers, function(marker) {
    allele_frequencies &lt;- df %&gt;%
      filter(marker == marker, population == population) %&gt;%
      pull(frequency) %&gt;%
      setNames(df$allele)
      
    num_shared_alleles[marker] / (sum(1 / allele_frequencies[simulated_genotypes$individual1[[marker]]]) + sum(1 / allele_frequencies[simulated_genotypes$individual2[[marker]]]))
  })
  
  # Return the index of relatedness
  return(list(relatedness = relatedness, num_shared_alleles = num_shared_alleles))
  }

# Function to simulate genotypes and calculate relatedness for different relationships for all markers
simulate_relatedness_all_markers &lt;- function(df, relationship_type, population) {
  # Simulate genotypes for all markers
  simulated_genotypes &lt;- simulate_genotypes_all_markers(df, relationship_type, population)
  
  # Calculate and return the relatedness for all markers
  relatedness_data &lt;- calculate_relatedness_all_markers(simulated_genotypes, df, population)
  
   # Before returning the results_df, add marker and population information
  results_df$marker &lt;- rownames(results_df)
  results_df$population &lt;- population
  
  return(relatedness_data)
}</code></pre>
<div id="visualizations" class="section level4">
<h4>Visualizations</h4>
<pre class="r"><code># Define the list of relationship types
relationship_types &lt;- c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)

# Create a dataframe of all combinations of populations and relationship types
combinations &lt;- expand.grid(population = unique(df$population), relationship_type = relationship_types)

# Apply the function to each combination
results &lt;- combinations %&gt;%
  split(seq(nrow(.))) %&gt;%
  map_dfr(function(combination) {
    population &lt;- combination$population
    relationship_type &lt;- combination$relationship_type
    # cat(&quot;Processing:&quot;, &quot;population =&quot;, population, &quot;; relationship_type =&quot;, relationship_type, &quot;\n&quot;)
    sim_results &lt;- simulate_relatedness_all_markers(df, relationship_type[[1]], population[[1]])
    
    # Bind resulting data frames
    tibble(
      population = population,
      relationship_type = relationship_type,
      sim_results = list(sim_results)
    )
  })

multi_results &lt;- results %&gt;% 
  mutate(
    sum_relatedness = map_dbl(sim_results, function(x) {
      sum(x[[&quot;relatedness&quot;]], na.rm = TRUE)
    }),
    sum_alleles = map_dbl(sim_results, function(x) {
      sum(x[[&quot;num_shared_alleles&quot;]], na.rm = TRUE)
    })
  ) %&gt;% 
  select(-sim_results)

multi_results</code></pre>
<pre><code># A tibble: 30 × 4
   population relationship_type sum_relatedness sum_alleles
   &lt;fct&gt;      &lt;fct&gt;                       &lt;dbl&gt;       &lt;dbl&gt;
 1 all        parent_child                0.339          32
 2 AfAm       parent_child                0.255          31
 3 Cauc       parent_child                0.142          34
 4 Hispanic   parent_child                0.166          30
 5 Asian      parent_child                0.313          33
 6 all        full_siblings               0.339          35
 7 AfAm       full_siblings               0.325          39
 8 Cauc       full_siblings               0.401          39
 9 Hispanic   full_siblings               0.306          25
10 Asian      full_siblings               0.223          27
# ℹ 20 more rows</code></pre>
<pre class="r"><code># Define the number of simulations
n_sims &lt;- 10

# Define the list of relationship types
relationship_types &lt;- c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)

# Create a dataframe of all combinations of populations, relationship types, and simulations
combinations &lt;- expand.grid(population = unique(df$population), relationship_type = relationship_types, simulation = 1:n_sims)

# Apply the function to each combination
results &lt;- combinations %&gt;%
  split(seq(nrow(.))) %&gt;%
  map_dfr(function(combination) {
    population &lt;- combination$population
    relationship_type &lt;- combination$relationship_type
    sim &lt;- combination$simulation
    # cat(&quot;Processing:&quot;, &quot;population =&quot;, population, &quot;; relationship_type =&quot;, relationship_type, &quot;; simulation =&quot;, sim, &quot;\n&quot;)
    sim_results &lt;- simulate_relatedness_all_markers(df, relationship_type[[1]], population[[1]])
    
    # Bind resulting data frames
    tibble(
      population = population,
      relationship_type = relationship_type,
      simulation = sim,
      sim_results = list(sim_results)
    )
  })

multi_results &lt;- results %&gt;% 
  mutate(
    sum_relatedness = map_dbl(sim_results, function(x) {
      sum(x[[&quot;relatedness&quot;]], na.rm = TRUE)
    }),
    sum_alleles = map_dbl(sim_results, function(x) {
      sum(x[[&quot;num_shared_alleles&quot;]], na.rm = TRUE)
    })
  ) %&gt;% 
  select(-sim_results)

multi_results</code></pre>
<pre><code># A tibble: 300 × 5
   population relationship_type simulation sum_relatedness sum_alleles
   &lt;fct&gt;      &lt;fct&gt;                  &lt;int&gt;           &lt;dbl&gt;       &lt;dbl&gt;
 1 all        parent_child               1           0.581          35
 2 AfAm       parent_child               1           0.237          31
 3 Cauc       parent_child               1           0.239          31
 4 Hispanic   parent_child               1           0.263          32
 5 Asian      parent_child               1           0.224          29
 6 all        full_siblings              1           0.409          26
 7 AfAm       full_siblings              1           0.479          35
 8 Cauc       full_siblings              1           0.230          28
 9 Hispanic   full_siblings              1           0.280          26
10 Asian      full_siblings              1           0.305          32
# ℹ 290 more rows</code></pre>
<pre class="r"><code>create_plot &lt;- function(df, variable_to_plot) {
  # Create the plot
  p &lt;- ggplot(df, aes(x = relationship_type, y = .data[[variable_to_plot]], color = population, shape = population, fill = population)) +
    geom_boxplot(alpha = 0.4, position = position_dodge(width = 0.75)) +
    geom_point(position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75), size = 1, alpha = 0.6) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    scale_x_discrete(limits = c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)) +
    labs(title = paste(ucfirst(variable_to_plot), &quot;by Relationship Type and Population&quot;),
         x = &quot;Relationship Type&quot;, 
         y = ucfirst(variable_to_plot), 
         color = &quot;Population&quot;,
         shape = &quot;Population&quot;)
  
  return(p)
}</code></pre>
<pre class="r"><code># Filter your data for unique STR markers and remove the &quot;all&quot; population
df_plt_multi_results &lt;- multi_results %&gt;%
  filter(population != &quot;all&quot;) %&gt;%
  select(-simulation)

p &lt;- create_plot(df_plt_multi_results, &quot;sum_relatedness&quot;)
p</code></pre>
<p><img src="figure/CODIS-STR-freqs.Rmd/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/unnamed-chunk-5-1.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plt_allele &lt;- create_plot(df_plt_multi_results, &quot;sum_alleles&quot;)

plt_allele</code></pre>
<p><img src="figure/CODIS-STR-freqs.Rmd/unnamed-chunk-5-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/cf281b698161b692f571b83e0d7a11f828eb6a91/docs/figure/CODIS-STR-freqs.Rmd/unnamed-chunk-5-2.png" target="_blank">cf281b6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>`</p>
</div>
</div>
</div>
<div id="updated-lr" class="section level2">
<h2>Updated LR</h2>
<div id="from-weight-of-evidence-for-forensic-dna-profiles-book"
class="section level3">
<h3>From Weight-of-evidence for forensic DNA profiles book</h3>
<p>Likelihood ratio for a single locus is:</p>
<p><span class="math display">\[
R=\kappa_0+\kappa_1 / R_X^p+\kappa_2 / R_X^u
\]</span> Where <span class="math inline">\(\kappa\)</span> is the
probability of having 0, 1 or 2 alleles IBD for a given
relationship.</p>
<p>The <span class="math inline">\(R_X\)</span> terms are quantifying
the “surprisingness” of a particular pattern of allele sharing.</p>
<p>The <span class="math inline">\(R_X^p\)</span> terms attached to the
<span class="math inline">\(kappa_1\)</span> are defined in the
following table:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\text { Table 7.2 Single-locus LRs for paternity when }
\mathcal{C}_M \text { is unavailable. }\\
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \times\left(1+2 F_{S T}\right) \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; 3 F_{S T}+\left(1-F_{S
T}\right) p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right)\left(F_{S T}+\left(1-F_{S T}\right) p_B\right)
/\left(2 F_{S T}+\left(1-F_{S T}\right)\left(p_A+p_B\right)\right) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>For our purposes we will take out the <span
class="math inline">\(F_{S T}\)</span> values. So the table will be as
follows:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2 p_A \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2p_A \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4p_A \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4 p_A p_B/(p_A+p_B) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>If none of the alleles match, then the <span
class="math inline">\(\kappa_1 / R_X^p = 0\)</span>.</p>
<p>The <span class="math inline">\(R_X^u\)</span> terms attached to the
<span class="math inline">\(kappa_2\)</span> are defined as:</p>
<p>If both alleles match and are homozygous the equation is 6.4 (pg 85).
Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AA}\)</span>
<span class="math display">\[
\frac{\left(2 F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(3 F_{S
T}+\left(1-F_{S T}\right) p_A\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to: <span class="math display">\[
p_A{ }^2
\]</span></p>
<p>If both alleles match and are heterozygous, the equation is 6.5 (pg
85) Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AB}\)</span>
<span class="math display">\[
2 \frac{\left(F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(F_{S
T}+\left(1-F_{S T}\right) p_B\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to:</p>
<p><span class="math display">\[
2 p_A p_B
\]</span> If both alleles do not match then <span
class="math inline">\(\kappa_2 / R_X^u = 0\)</span>.</p>
</div>
<div id="calculating-lr" class="section level3">
<h3>Calculating LR</h3>
<p>We need 100,000 unrelated pairs.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.7.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.4.0  knitr_1.45        data.table_1.15.0 lubridate_1.9.3  
 [5] forcats_1.0.0     stringr_1.5.1     dplyr_1.1.4       purrr_1.0.2      
 [9] readr_2.1.5       tidyr_1.3.1       tibble_3.2.1      ggplot2_3.4.4    
[13] tidyverse_2.0.0   readxl_1.4.3      workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.41         bslib_0.6.1       processx_3.8.3   
 [5] callr_3.7.3       tzdb_0.4.0        vctrs_0.6.5       tools_4.3.2      
 [9] ps_1.7.6          generics_0.1.3    fansi_1.0.6       highr_0.10       
[13] pkgconfig_2.0.3   rematch_2.0.0     lifecycle_1.0.4   compiler_4.3.2   
[17] farver_2.1.1      git2r_0.33.0      munsell_0.5.0     getPass_0.2-4    
[21] httpuv_1.6.14     htmltools_0.5.7   sass_0.4.8        yaml_2.3.8       
[25] later_1.3.2       pillar_1.9.0      jquerylib_0.1.4   whisker_0.4.1    
[29] cachem_1.0.8      tidyselect_1.2.1  digest_0.6.34     stringi_1.8.3    
[33] labeling_0.4.3    rprojroot_2.0.4   fastmap_1.1.1     grid_4.3.2       
[37] colorspace_2.1-0  cli_3.6.2         magrittr_2.0.3    utf8_1.2.4       
[41] withr_3.0.0       scales_1.3.0      promises_1.2.1    timechange_0.3.0 
[45] rmarkdown_2.25    httr_1.4.7        cellranger_1.1.0  hms_1.1.3        
[49] evaluate_0.23     viridisLite_0.4.2 rlang_1.1.3       Rcpp_1.0.12      
[53] glue_1.7.0        xml2_1.3.6        svglite_2.1.3     rstudioapi_0.15.0
[57] jsonlite_1.8.8    R6_2.5.1          systemfonts_1.0.5 fs_1.6.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
