<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sam Muller" />


<title>Likelihood Ratio</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lasisilab/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Likelihood Ratio</h1>
<h4 class="author">Sam Muller</h4>
<h4 class="date">2024-09-16 20:56:47</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree2479f41e092c583d03a0ac0ebf554c0bfd60625etargetblank2479f41a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/lasisilab/PODFRIDGE/tree/2479f41e092c583d03a0ac0ebf554c0bfd60625e" target="_blank">2479f41</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomlasisilabPODFRIDGEtree2479f41e092c583d03a0ac0ebf554c0bfd60625etargetblank2479f41a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/lasisilab/PODFRIDGE/tree/2479f41e092c583d03a0ac0ebf554c0bfd60625e" target="_blank">2479f41</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  data/final_CODIS_data.csv

Unstaged changes:
    Deleted:    analysis/CODIS_DB_composition.Rmd
    Deleted:    analysis/IBD.Rmd
    Deleted:    analysis/STR.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/analyses.Rmd
    Deleted:    analysis/background.Rmd
    Deleted:    analysis/demo-data.Rmd
    Modified:   analysis/final_equation.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Deleted:    analysis/manuscript_figures.Rmd
    Modified:   analysis/regression.Rmd
    Deleted:    analysis/representation-risk.Rmd
    Deleted:    slurm-11226865.out
    Deleted:    slurm-11226959.out
    Deleted:    slurm-11227723.out
    Deleted:    slurm-11228488.out
    Deleted:    slurm-11228791.out
    Deleted:    slurm-11229998.out
    Deleted:    slurm-11230240.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/lr.Rmd</code>) and HTML
(<code>docs/lr.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/9a85666490b7d01f6566d3c038676d8e95182038/docs/lr.html" target="_blank">9a85666</a>
</td>
<td>
sammuller917
</td>
<td>
2024-07-11
</td>
<td>
Update lr.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/9ce56263ac96bf626e274bd67beff7283fbd76e4/analysis/lr.Rmd" target="_blank">9ce5626</a>
</td>
<td>
sammuller917
</td>
<td>
2024-07-11
</td>
<td>
Update lr.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/49f9b23e3f37103616b1d99f6b2654f3206127ff/docs/lr.html" target="_blank">49f9b23</a>
</td>
<td>
sammuller917
</td>
<td>
2024-07-03
</td>
<td>
created lr page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/8f1347cb4ed22c3890320a33a46c084c1b07b6b3/analysis/lr.Rmd" target="_blank">8f1347c</a>
</td>
<td>
sammuller917
</td>
<td>
2024-07-03
</td>
<td>
Update lr.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lasisilab/PODFRIDGE/b71e11f832df192746780574bf1647081c8c0efe/docs/lr.html" target="_blank">b71e11f</a>
</td>
<td>
linmatch
</td>
<td>
2024-07-03
</td>
<td>
new com
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/609a24074616215781a2332c8a0471baf6f3b337/analysis/lr.Rmd" target="_blank">609a240</a>
</td>
<td>
sammuller917
</td>
<td>
2024-07-03
</td>
<td>
Update lr.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lasisilab/PODFRIDGE/blob/009643cee337ea7c1e1e739fe9be1f78dc980624/analysis/lr.Rmd" target="_blank">009643c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-07-01
</td>
<td>
Create lr.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>5.6 in Weight of Evidence The following equation predicts the
probability that the next allele sampled in a population will be allele
A, also known as the Match Probability, <span
class="math inline">\(M\)</span>. <span class="math display">\[
M=\frac{m \theta+(1-\theta) p}{1+(n-1) \theta}
\]</span> Where <span class="math inline">\(p\)</span> is the
probability of allele A appearing in the population, <span
class="math inline">\(n\)</span> is the number of alleles being sampled,
and <span class="math inline">\(m\)</span> is the number of observed
allele A in the sampled population.</p>
<p><span class="math inline">\(\theta\)</span> represents the correction
for population stratification and genetic drift. The higher the value of
<span class="math inline">\(\theta\)</span>, the higher the likelihood
of seeing any given allele in a population once it has already been
observed. The most common <span class="math inline">\(\theta\)</span>
value is 0.01, but in many cases <span
class="math inline">\(\theta\)</span> can be set to 0.</p>
</div>
<div id="developing-the-likelihood-ratio" class="section level2">
<h2>Developing the Likelihood Ratio</h2>
<p>A likelihood ratio calculation will be used to compare a forensic
unknown, profile c, to a convicted offender profile, profile O, and
determine the likelihood that those individuals are related. To
determine the likelihood of relatedness, or the relatedness score, we
compare the probability that profile A and profile B sharing alleles at
specific loci is due to them being related, versus the probability that
the shared alleles are due to random chance and the pair is unrelated.
This gives us our base likelihood ratio, which we will be deriving using
the equation above. <span class="math display">\[
R = \frac{P(profile\:c
\:|\:profile\:O,\:related)}{P(profile\:c\:|\:profile\:O,\:unrelated)}
\]</span> We know related individuals will likely share a certain number
of alleles due to recent common ancestors. This is known as alleles
being identical by descent, or IBD, and can either be 0, 1, or 2. The
probability for sharing 0, 1, or 2 alleles due to IBD is represented as
<span class="math inline">\(\kappa\)</span>, and will be calculated for
each relationship as shown in the table below.</p>
<p>table</p>
<p>To determine the probability of sharing <span
class="math inline">\(i\)</span> alleles for a related pair, we must
multiple the probability of sharing 0, 1, or 2 alleles due to IBD by the
probability of profile O having <span class="math inline">\(i\)</span>
shared alleles based on profile c, and sum the possible options for all
values of i.</p>
<p>This results in the following equation: <span class="math display">\[
P(profile\:c\:|\:profile\:O,\:related)=\kappa_2(M_0)+\kappa_1(M_1)+\kappa_0(M_2)
\]</span></p>
<p>In the instance of <span class="math inline">\(\kappa_2\)</span>, we
know the match probability to be certain, so <span
class="math inline">\(M_0 = 1\)</span>, as in this scenario there is a
100% match between both alleles due to them both being IBD. If both
alleles in profile O do not match to profile c, then <span
class="math inline">\(\kappa_2\)</span> will be 0 and will not be
used.</p>
<p>To determine what equations should be used for <span
class="math inline">\(M_1\)</span> and <span
class="math inline">\(M_2\)</span>, we must look at the alleles of the
population being sampled, that is, the alleles of profile c and profile
O at the designated locus. The table below describes the potential
allele combinations at a designated locus where C is any other allele.
<span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; O \\
\hline \mathrm{AA} &amp; \mathrm{AA} \\
\mathrm{AA} &amp; \mathrm{AB} \\
\mathrm{AB} &amp; \mathrm{AA}\\
\mathrm{AB} &amp; \mathrm{AB}\\
\mathrm{AB} &amp; \mathrm{AC} \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p><span class="math inline">\(M_1\)</span> will tell us the match
probability of profile O to <strong>one</strong> allele in profile c. To
calculate <span class="math inline">\(M_1\)</span> we must evaluate the
scenarios in which profile c is homozygous or heterozygous.</p>
<p>In the case of homozygous profile c (AA) we calculate the probability
that profile O has a matching allele A. This is based on the condition
that two A alleles have already been observed in the sampled population,
giving us an <span class="math inline">\(m\)</span> value of 2 and an
<span class="math inline">\(n\)</span> value of 2. Using equation 5.6 we
come up with the following <span class="math inline">\(M_1\)</span> for
use when profile c is homozygous <span class="math display">\[
M_{1\:c|homo}=\frac {2\theta+(1-\theta)p_A}{1+(2-1)\theta}
\]</span> which when <span class="math inline">\(\theta = 0\)</span>
reduces to <span class="math inline">\(p_A\)</span></p>
<p>In the case of heterozygous profile c (AB) we calculate the
probability that profile O has one matching allele, either matching
allele A or allele B. This will sum the equation for observing an A
allele after one has already been observed, <span
class="math inline">\(m = 1\)</span>, <span class="math inline">\(n =
2\)</span>, and <span class="math inline">\(p_A\)</span>, with the
equation for observing a B allele after one has already been observed,
<span class="math inline">\(m = 1\)</span>, <span
class="math inline">\(n = 2\)</span>, <span
class="math inline">\(p_B\)</span>. The options are equally likely, so
we multiply by <span class="math inline">\(\frac {1}{2}\)</span>. Using
equation 5.6 we come up with the following <span
class="math inline">\(M_1\)</span> for use when profile c is
heterozygous.<br />
<span class="math display">\[
\begin{equation}
M_{1\:c|hetero}=(\frac {1\theta+(1-\theta)p_A}{1+(2-1)\theta}+\frac
{1\theta+(1-\theta)p_B}{1+(2-1)\theta})*\frac{1}{2}=\frac
{2\theta+(1-\theta)(p_A+p_B)}{2+2\theta}*\frac{1}{2}\\
\: \\
M_{1\:c|hetero}=\frac{\theta+(1-\theta)(p_A+p_B)/2}{1+\theta}
\end{equation}
\]</span> which when <span class="math inline">\(\theta = 0\)</span>
reduces to <span class="math inline">\(\frac{p_A+p_B}{2}\)</span></p>
<p>In the scenario of <span class="math inline">\(\kappa_0\)</span>, no
alleles are presumed IBD, so we must use <span
class="math inline">\(M_2\)</span> to determine the probability that the
alleles in profile O match to any of the alleles found in profile c
randomly. This is also known as the unrelated match probability.</p>
<p>To determine <span class="math inline">\(M_2\)</span>, we again start
with profile c, and ask whether it is homozygous or heterozygous.</p>
<p>If profile c is homozygous, we are trying to determine the
probability of profile O having matching two matching allele As. This
probability is the product of the calculation for each allele in profile
O given that all previous alleles have been A. We will calculate
equation 5.6 using an m and n value of 2, and then an m and n value of
3, as seen below, or in the Weight of Evidence as formula 6.3 <span
class="math display">\[
M_{2\:
homo}=\frac{2\theta+(1-\theta)p_A}{1+(2-1)\theta}*\frac{3\theta+(1-\theta)p_A}{1+(3-1)\theta}
\]</span> which when <span class="math inline">\(\theta = 0\)</span>
reduces to <span class="math inline">\({p_A}^2\)</span></p>
<p>In the instance where only one allele matches between profile A and
profile C, <span class="math inline">\(M_2\)</span> would be doubled to
account for either allele being the one that is IBD vs IBS, making <span
class="math inline">\(M_2 = 2{p_A}^2\)</span>.</p>
<p>If profile c is heterozygous, we are trying to determine the
probability of profile O having two matching alleles, allele A and
allele B. This probability evaluates 5.6 for the instances where m = 1,
n = 2 and again with m = 1, n = 3, and multiplying by two for the two
possible orderings of the A and B alleles, as seen below and in Weight
of Evidence as formula 6.4 <span class="math display">\[
M_{2\:
hetero}=2\frac{\theta+(1-\theta)p_A}{1+(2-1)\theta}*\frac{\theta+(1-\theta)p_B}{1+(3-1)\theta}
\]</span> which when <span class="math inline">\(\theta = 0\)</span>
reduces to <span class="math inline">\(2p_Ap_B\)</span></p>
<p>The various match probability equations for each scenario are shown
in the table below: <span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; O &amp; \kappa_i\:used &amp;
M_0|\theta=0&amp;M_1|\theta=0&amp;M_2|\theta=0\\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; \kappa_2,\kappa_1,\kappa_0
&amp; 1&amp;p_A &amp; {p_A}^2 \\
\mathrm{AA}&amp; \mathrm{AC} &amp; \kappa_1,\kappa_0 &amp; 0 &amp; p_A
&amp; 2{p_A}^2\\
\mathrm{AB} &amp; \mathrm{AC} &amp; \kappa_1,\kappa_0 &amp; 0 &amp;
\frac{p_A}{2} &amp; {p_A}^2 \\
\mathrm{AB} &amp; \mathrm{CB} &amp; \kappa_1,\kappa_0 &amp; 0 &amp;
\frac{p_B}{2} &amp; {p_B}^2 \\
\mathrm{AB} &amp; \mathrm{AB} &amp; \kappa_2,\kappa_1,\kappa_0 &amp; 1
&amp; \frac{p_A}{2}+\frac{p_B}{2} &amp; 2p_Ap_B \\
\hline
\end{array}
\end{aligned}
\]</span> ## The Final Likelihood Ratio To calculate the final
likelihood ratio, we must place our related probability estimate for
matched alleles over the unrelated estimate for matched alleles. We have
already done the calculation for this denominator in calculating the
<span class="math inline">\(M_2\)</span> equation, which is comparing
the probability of a profile appearing twice in a pair of unrelated
individuals</p>
<p>For heterozygous profile c using a <span
class="math inline">\(\theta=0\)</span>, the denominator of the
likelihood ratio will be <span
class="math inline">\(2p_Ap_B\)</span></p>
<p>For homozygous profile c using a <span
class="math inline">\(\theta=0\)</span>, the denominator of the
likelihood ratio will be <span
class="math inline">\({p_A}^2\)</span></p>
<p>The following proofs demonstrate the simplification of the likelihood
ratio equations for each scenario present in the table</p>
<p><span class="math display">\[
\begin{equation}
profile\;c=AA,\;profile\;O=AA\\
R=\frac{\kappa_2+\kappa_1(p_A)+\kappa_0({p_A}^2)}{{p_A}^2}=\frac{\kappa_2}{{p_A}^2}+\frac{\kappa_1}{p_A}+\kappa_0
\end{equation}
\]</span> <span class="math display">\[
\begin{equation}
profile\;c=AA,\;profile\;O=AC\\
R=\frac{\kappa_1(p_A)+\kappa_0(2{p_A}^2)}{2{p_A}^2}=\frac{\kappa_1}{2p_A}+\kappa_0
\end{equation}
\]</span> <span class="math display">\[
\begin{equation}
profile\;c=AB,\;profile\;O=AC\\
R=\frac{\kappa_1(\frac{p_A}{2})+\kappa_0({p_A}^2)}{{p_A}^2}=\frac{\kappa_1}{2p_A}+\kappa_0
\end{equation}
\]</span> This above equation would also be utilized in the instance of
profile c = AB and profile O = CB, although <span
class="math inline">\(p_B\)</span> would be substituted for <span
class="math inline">\(p_A\)</span>.</p>
<p><span class="math display">\[
\begin{equation}
profile\;c=AB,\;profile\;O=AB\\
R=\frac{\kappa_2+\kappa_1(\frac{p_A}{2}+\frac{p_B}{2})+\kappa_0(2p_Ap_B)}{2p_Ap_B}=\frac{\kappa_2}{2p_Ap_B}+\frac{\kappa_1}{\frac{p_A+p_B}{4p_Ap_B}}+\kappa_0
\end{equation}
\]</span></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.3         knitr_1.48       
 [5] rlang_1.1.4       xfun_0.45         stringi_1.8.4     processx_3.8.4   
 [9] promises_1.3.0    jsonlite_1.8.8    glue_1.7.0        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.8.1 httpuv_1.6.15     ps_1.7.7         
[17] sass_0.4.9        fansi_1.0.6       rmarkdown_2.27    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.24.0   fastmap_1.2.0     yaml_2.3.9       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1     compiler_4.4.1   
[29] fs_1.6.4          pkgconfig_2.0.3   Rcpp_1.0.12       rstudioapi_0.16.0
[33] later_1.3.2       digest_0.6.36     R6_2.5.1          utf8_1.2.4       
[37] pillar_1.9.0      callr_3.7.6       magrittr_2.0.3    bslib_0.7.0      
[41] tools_4.4.1       cachem_1.1.0      getPass_0.2-4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
