<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Methods</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="demo-data.html">Demographic Data</a>
</li>
<li>
  <a href="database-composition.html">Database composition</a>
</li>
<li>
  <a href="STR-simulation.html">STR simulations</a>
</li>
<li>
  <a href="CODIS-STR-freqs.html">CODIS STRs</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tinalasisi/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Methods</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2024-03-03 20:03:31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtree2596546a55e9369c81787cb7f9f35c660272aa16targetblank2596546a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/2596546a55e9369c81787cb7f9f35c660272aa16" target="_blank">2596546</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtree2596546a55e9369c81787cb7f9f35c660272aa16targetblank2596546a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/2596546a55e9369c81787cb7f9f35c660272aa16" target="_blank">2596546</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    output/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/methods.Rmd</code>) and HTML
(<code>docs/methods.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/48acb9f6ede345fe2546bcf9bff6264624ab9e88/docs/methods.html" target="_blank">48acb9f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/aa3ff5cd58cd98d8f1274294dfd38b244f01c65b/docs/methods.html" target="_blank">aa3ff5c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/9e71347bf88281897c8f83d0531c703b068b0267/docs/methods.html" target="_blank">9e71347</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/6e2b61a790a03ec3235088f637bd118f71448535/analysis/methods.Rmd" target="_blank">6e2b61a</a>
</td>
<td>
GitHub
</td>
<td>
2023-06-27
</td>
<td>
singular
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/c280b6f940af31573e4c3476b16188af1604f146/docs/methods.html" target="_blank">c280b6f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/f89a90f2c7cc6808f21d805b3ba5df810ed06184/docs/methods.html" target="_blank">f89a90f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/dbad021423e088080ae2981dd213b0e26fa82c84/analysis/methods.Rmd" target="_blank">dbad021</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-11
</td>
<td>
Update methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/dbad021423e088080ae2981dd213b0e26fa82c84/docs/methods.html" target="_blank">dbad021</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-11
</td>
<td>
Update methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/2a436aab6857b77db689b72aac35dbdcd5027f94/docs/methods.html" target="_blank">2a436aa</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/97753d69ffbe2865c3da8a91b5003ec5cd4e1f6c/analysis/methods.Rmd" target="_blank">97753d6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-10
</td>
<td>
wflow_publish("analysis/methods.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/d97facec79dc362c30390891b60482d208807811/docs/methods.html" target="_blank">d97face</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/20f1d6cb96d8c148a47462042655be95461c97e0/analysis/methods.Rmd" target="_blank">20f1d6c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-10
</td>
<td>
wflow_publish("analysis/methods.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/f4a00f40edb46414164a7754bdc3fe17479a6ff7/analysis/methods.Rmd" target="_blank">f4a00f4</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-08
</td>
<td>
Typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/799679a0f2410722c25f245d40762323bd626c3a/docs/methods.html" target="_blank">799679a</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/249447bef9db3cf63044cfe6979729c71c42d32b/analysis/methods.Rmd" target="_blank">249447b</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-27
</td>
<td>
questions in original code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/627214dca182e09f02ff5718455ed5fcccbbd788/docs/methods.html" target="_blank">627214d</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-23
</td>
<td>
Update methods.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/6605057b54542682b623705449b2bc0b28bca277/analysis/methods.Rmd" target="_blank">6605057</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-20
</td>
<td>
Update parameters for generations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/5f805fe9b5638c72575d8adca031aa08ee3e9a31/docs/methods.html" target="_blank">5f805fe</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/e912cb8c6ea352be3c54af4c674c47697d67ba93/analysis/methods.Rmd" target="_blank">e912cb8</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-06
</td>
<td>
Updating methods and adding data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/37a3d63e27a403d1c3b3e83a8521f6d24a6a81b0/docs/methods.html" target="_blank">37a3d63</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/c02c42713c8a5a0e7d87c85aba2eb1c601691e70/analysis/methods.Rmd" target="_blank">c02c427</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
update methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/342c5a332d216f320b437c315d1c394ce83456a2/docs/methods.html" target="_blank">342c5a3</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/ea3444ea181e5acc74785bcf8102b65cda3d05da/analysis/methods.Rmd" target="_blank">ea3444e</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
update methods
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/2b161f2abfe670baa9826c8a1f4857f376c768ab/docs/methods.html" target="_blank">2b161f2</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/d8748aa57fc64e442e3bfd4dbd480abd67ce1d5d/analysis/methods.Rmd" target="_blank">d8748aa</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
wflow_publish("analysis/methods.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/c3948af0114e77dacaf5bd4221c6b4639c120655/docs/methods.html" target="_blank">c3948af</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/ce722bc8d9c60115d8a82c053a644774647ba6f7/analysis/methods.Rmd" target="_blank">ce722bc</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
<td>
update website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/a3af2d6117b4dfdb4d007d0f6ed7d04a58711883/analysis/methods.Rmd" target="_blank">a3af2d6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Updating images and methods page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/a3af2d6117b4dfdb4d007d0f6ed7d04a58711883/docs/methods.html" target="_blank">a3af2d6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Updating images and methods page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/f02bc38a894a24323df5f6dadaf096826a830b88/docs/methods.html" target="_blank">f02bc38</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/c9130d5e7bf0085f4fd6269628f455f21bbfbc65/docs/methods.html" target="_blank">c9130d5</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/a4a7d45ed9fa00ec06842c882a5e6b6803a72eba/docs/methods.html" target="_blank">a4a7d45</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/00073fd8cef2219c375fbfdd0bc41402d030b5a1/docs/methods.html" target="_blank">00073fd</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/d6a84dfc93c0f9c3512830250138ad851d77f2eb/analysis/methods.Rmd" target="_blank">d6a84df</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-03
</td>
<td>
Update website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/51ed5a6b8f5bc6d2bb799e9015da02662d74e21c/docs/methods.html" target="_blank">51ed5a6</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/13ed9aecb20d1ca7b4c9dcbd6f1949f15ee1e341/analysis/methods.Rmd" target="_blank">13ed9ae</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-02
</td>
<td>
Publishing POPFORGE
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/13ed9aecb20d1ca7b4c9dcbd6f1949f15ee1e341/docs/methods.html" target="_blank">13ed9ae</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-02
</td>
<td>
Publishing POPFORGE
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="long-range-familial-searching" class="section level1">
<h1>Long-Range Familial Searching</h1>
<div id="coop-edge-2019" class="section level2">
<h2>Coop &amp; Edge 2019</h2>
<p>Coop &amp; Edge (<a
href="https://www.biorxiv.org/content/10.1101/531384v1.full">2019</a>)
use a population genetics approach to estimate the likelihood of finding
a genetic relative in a database. They assume that two individuals are
related if they share a certain number of genetic blocks, which is
determined by the degree of relatedness (e.g. first cousin, second
cousin, etc.). Using this assumption, they calculate the expected number
of blocks shared between two individuals of a given degree of
relatedness based on the size of the genetic database and the population
size. They then uses the Poisson distribution to estimate the
probability of finding at least one relative of a given degree of
relatedness in a database of a certain size.</p>
<p>The original code can be found <a
href="https://github.com/tinalasisi/POPFORGE/blob/master/data/Coop_cousins_vs_database_size.R">here</a>.</p>
<p>Below is an adaptation of the code for demonstration purposes with
annotations for clarification.</p>
<div id="population-and-database-size-estimation"
class="section level3">
<h3>Population and Database Size Estimation</h3>
<p>In the code chunk below, we make several assumptions about the
population size of possible ancestors based on data from the US census.
The assumptions used in the analysis include an average time of 30 years
between generations, and a population size for the relevant time period
calculated as the size of the population in 1950, adjusted for the
number of generations separating the individuals in question.</p>
<p>Additionally, it is assumed that about half of individuals in the
population are potentially parents, and 90% of those individuals have
children. The minimum population size is set to 1 million for
populations smaller than that, and a vector of database sizes is created
to test the expected number of cousins in a database of various
sizes.</p>
<pre class="r"><code># Load necessary packages
library(wesanderson) # for color palettes

# Set path to the data file
path &lt;- &quot;./data/&quot;
savepath &lt;- &quot;./output/&quot;

# Set up vector for cousin degree
p &lt;- c(1:8) 

# Set up initial population size
N &lt;- 76e6 

# Read in data on US population sizes by year
US_pop &lt;- read.csv(paste(path, &quot;US_popsize.csv&quot;, sep = &quot;&quot;))

# Calculate number of grandparents by generation
p_grandpar_gen &lt;- 1950 - 30 * (p + 1)

# Determine which years match the grandparents&#39; generation
these.years &lt;- match(p_grandpar_gen, US_pop$Year)

# Get population sizes by year for grandparents&#39; generation
US_Ns &lt;- US_pop$Population[these.years]

# Scale population size down by 50% (assumed number of potential parents) and 90% of those have children 
N &lt;- US_Ns
N &lt;- N * 0.5 * 0.9 

# Set minimum population size for small populations
N[US_Ns &lt; 1e6] &lt;- 1e6 # is this in the correct order? 

# Set up vector of database sizes to test
DB.sizes &lt;- c(1e6, 5e6, 10e6)

# Set color palette for graphs
my.cols &lt;- wes_palette(&quot;Darjeeling1&quot;)</code></pre>
</div>
<div
id="probability-of-p-th-degree-cousin-and-number-of-p-th-degree-cousins"
class="section level3">
<h3>Probability of p-th degree cousin and number of p-th degree
cousins</h3>
<p>The code below calculates the probability of having at least one p-th
cousin in the database and the expected number of p-th cousins in the
sample for different database sizes. The first plot shows the
probability of having at least one p-th cousin in the database based on
the degree of relatedness and the size of the database. The second plot
shows the expected number of p-th cousins in the sample for different
database sizes.</p>
<pre class="r"><code>layout(t(1:2))

# Plot probability of having at least one p-th cousin in database
plot(c(1, 8), c(0, 1), type = &quot;n&quot;, ylab = &quot;Probability of at least one p-th cousin in database&quot;, xlab = &quot;p (degree of cousin)&quot;)
sapply(1:length(DB.sizes), function(i) {
  DB.size &lt;- DB.sizes[i]
  prob.no.rellys &lt;- exp(-2^(2*p - 2) * DB.size / N)
  points(p, 1 - prob.no.rellys, type = &quot;b&quot;, col = my.cols[i], pch = 19, lwd = 1.5)
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<pre class="r"><code>legend(x = &quot;bottomright&quot;, legend = c(&quot;Database size (Millions) =&quot;, format(DB.sizes / 1e6, dig = 1)), col = c(NA, my.cols), pch = 19)

# Plot of expected number of p-th cousins in sample
plot(c(1,8),c(0,1000),type=&quot;n&quot;,ylab=&quot;Expected number of p-th cousins in database&quot;,xlab=&quot;p.(degree of cousin)&quot;)
sapply(1:length(DB.sizes),function(i){

    num.cousins&lt;-4^(p)*DB.sizes[i]/(N/2)
    points(p,num.cousins,type=&quot;b&quot;,col=my.cols[i],lwd=1.5,pch=19)
   # points(p,4^(p),type=&quot;b&quot;,col=&quot;black&quot;)
})</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/methods.Rmd/plt-cousins-1.png" alt="The plot above displays the probability of having at least one p-th cousin in a genetic database and the expected number of p-th cousins in a database. The left plot shows the probability of finding at least one p-th cousin in a database of a given size, and the right plot shows the expected number of p-th cousins in a database of a given size based on the number of genetic blocks shared." width="768" />
<p class="caption">
The plot above displays the probability of having at least one p-th
cousin in a genetic database and the expected number of p-th cousins in
a database. The left plot shows the probability of finding at least one
p-th cousin in a database of a given size, and the right plot shows the
expected number of p-th cousins in a database of a given size based on
the number of genetic blocks shared.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-cousins-1">
Past versions of plt-cousins-1.png
</button>
</p>
<div id="fig-plt-cousins-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/342c5a332d216f320b437c315d1c394ce83456a2/docs/figure/methods.Rmd/plt-cousins-1.png" target="_blank">342c5a3</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
</div>
<div id="probability-of-a-genetically-detectable-cousin"
class="section level3">
<h3>Probability of a genetically detectable cousin</h3>
<p>Below, we calculate the expected number of shared blocks of genetic
material between cousins of varying degrees of relatedness. This is
important because the probability of detecting genetic material that is
shared between two individuals decreases as the degree of relatedness
between them decreases. The code uses a Poisson distribution assumption
to estimate the probability of two cousins sharing at least one, two, or
three blocks of genetic material, based on the expected number of shared
blocks of genetic material calculated from previous research.</p>
<pre class="r"><code># The variable &#39;meiosis&#39; represents the number of meiosis events between cousins, where &#39;p&#39; is the degree of relatedness (i.e. p = 1 for first cousins, p = 2 for second cousins, etc.)
meiosis &lt;- p + 1

## Expected number of blocks shared between cousins
# &#39;E.num.blocks&#39; is the expected number of blocks of shared genetic material between cousins based on the degree of relatedness and the number of meiosis events between them. This value is calculated based on previous research and is not calculated in this code.
E.num.blocks &lt;- 2 * (33.8 * (2 * meiosis) + 22) / (2^(2 * meiosis - 1))

## Use Poisson assumption
# &#39;Prob.genetic&#39; is the probability of two cousins sharing at least one block of genetic material based on the expected number of shared blocks calculated in the previous step. The calculation uses a Poisson distribution assumption.
Prob.genetic &lt;- 1 - exp(-E.num.blocks)

# &#39;prob.g.e.2.blocks&#39; is the probability of two cousins sharing at least two blocks of genetic material based on the expected number of shared blocks calculated in the previous step. The calculation uses a Poisson distribution assumption.
prob.g.e.2.blocks &lt;- 1 - sapply(E.num.blocks, function(expected.num) {sum(dpois(0:1, expected.num))})

# &#39;prob.g.e.3.blocks&#39; is the probability of two cousins sharing at least three blocks of genetic material based on the expected number of shared blocks calculated in the previous step. The calculation uses a Poisson distribution assumption.
prob.g.e.3.blocks &lt;- 1 - sapply(E.num.blocks, function(expected.num) {sum(dpois(0:2, expected.num))})</code></pre>
<pre class="r"><code>## Plot for number of shared blocks with p-th degree cousins
# Set layout of plot
layout(t(1))

# Set color palette for plot
my.cols2&lt;-wes_palette(&quot;FantasticFox1&quot;)[3:5]

# Create a blank plot with labeled axes
plot(c(1,8),c(0,1),type=&quot;n&quot;,ylab=&quot;Probability p-th cousin \&quot;detectable\&quot;&quot;,xlab=&quot;p.(degree of cousin)&quot;)

# Add points for probability of detecting pth cousin with genomic blocks using colors from my.cols2
points(p,Prob.genetic,col=my.cols2[1],pch=19,type=&quot;b&quot;,lwd=2)
points(p,prob.g.e.2.blocks,col=my.cols2[2],pch=19,type=&quot;b&quot;,lwd=2)
points(p,prob.g.e.3.blocks,col=my.cols2[3],pch=19,type=&quot;b&quot;,lwd=2)

# Add a legend to the plot
legend(x=&quot;topright&quot;,legend=c(&quot;Cousins (w. &gt;0 genomic blocks)&quot;,&quot;Cousins (w. &gt;1 genomic blocks)&quot;,&quot;Cousins (w. &gt;2 genomic blocks)&quot;),col=my.cols2[1:3],lty=1)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/methods.Rmd/plt-genetic-blocks-1.png" alt="Probabilities of detecting a genetic cousin in a database based on shared genomic blocks. Blue lines represent cousins with at least one genomic block, orange dotted and red lines represent cousins with at least two and three genomic blocks, respectively. The legend specifies the type of cousin being represented by each line." width="672" />
<p class="caption">
Probabilities of detecting a genetic cousin in a database based on
shared genomic blocks. Blue lines represent cousins with at least one
genomic block, orange dotted and red lines represent cousins with at
least two and three genomic blocks, respectively. The legend specifies
the type of cousin being represented by each line.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-genetic-blocks-1">
Past versions of plt-genetic-blocks-1.png
</button>
</p>
<div id="fig-plt-genetic-blocks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/342c5a332d216f320b437c315d1c394ce83456a2/docs/figure/methods.Rmd/plt-genetic-blocks-1.png" target="_blank">342c5a3</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## Plot for expected number of p-th degree cousins in a given database

# Set the layout of the plot
layout(t(1))

# Create a plot with the y-axis ranging from 0 to 350 and the x-axis ranging from 1 to 8, labeled as &quot;Expected number of genetic p-th cousins in database&quot; and &quot;p. (degree of cousin)&quot;, respectively
plot(c(1,8),c(0,350),type=&quot;n&quot;,ylab=&quot;Expected number of genetic p-th cousins in database&quot;,xlab=&quot;p. (degree of cousin)&quot;)

# For each database size, calculate the expected number of genetic cousins based on the size of the database, the total number of individuals in the population, and the expected number of blocks of shared genetic material between cousins. Multiply the number of expected genetic cousins by the probability of sharing at least three blocks of genetic material with a cousin, and plot the result as a line on the plot. Repeat this for each database size.
sapply(1:length(DB.sizes),function(i){

    num.cousins&lt;-4^(p)*DB.sizes[i]/(N/2)
    points(p,num.cousins*prob.g.e.3.blocks,type=&quot;b&quot;,lty=1,col=my.cols[i])

})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<pre class="r"><code># Create a legend for the plot that explains what each line represents, including the legend for the database size
my.leg &lt;- c(c(&quot;Genetic Cousins&quot;, &quot;&gt;2 blocks&quot;), c(&quot;Database size (Millions)=&quot;, format(DB.sizes/1e6, dig=1)))
legend(x=&quot;topright&quot;,legend=my.leg,col=c(NA,rep(&quot;black&quot;,1),NA,my.cols),lty=c(NA,1:2,rep(NA,3)),pch=c(rep(NA,3),rep(19,3)))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/methods.Rmd/plt-num-cousins-1.png" alt="This figure shows the expected number of genetic relatives of degree p in a database of a given size. The x-axis represents the degree of relatedness, while the y-axis represents the expected number of genetic relatives. Each line represents a different database size." width="672" />
<p class="caption">
This figure shows the expected number of genetic relatives of degree p
in a database of a given size. The x-axis represents the degree of
relatedness, while the y-axis represents the expected number of genetic
relatives. Each line represents a different database size.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt-num-cousins-1">
Past versions of plt-num-cousins-1.png
</button>
</p>
<div id="fig-plt-num-cousins-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/342c5a332d216f320b437c315d1c394ce83456a2/docs/figure/methods.Rmd/plt-num-cousins-1.png" target="_blank">342c5a3</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="erlich-et-al-2018" class="section level2">
<h2>Erlich et al 2018</h2>
<p>The authors carried out a detailed investigation to find out how
likely it is to identify a person using long-range familial searches
with genetic data from direct-to-consumer databases. The goal was to
understand the privacy risks related to DNA data and its consequences
for law enforcement and research studies. To do this, the authors
created various mathematical models that considered factors like the
size of the population, the size of the database, and the criteria for
finding matches. They also took into account different types of
relationships, such as regular cousins and once-removed cousins.</p>
<p>The R code provided with the study includes functions that implement
these models and methods, which allows users to assess the likelihood of
identifying a person through long-range familial searches. The authors
used population genetics models to estimate the probabilities of finding
relatives with specific amounts of shared DNA, known as identity by
descent (IBD) segments. These models were based on assumptions like
uniform population growth, no inbreeding, and random sampling, providing
a rough guideline for estimating identification probabilities.</p>
<pre class="r"><code># Define genome size and number of chromosomes
genome_size = 35 
num_chrs = 22

# Function to calculate the probability of a match for direct relatives
p_match = function(g, m, min_num_seg)
{
  m = m / 100 # Convert m from cM to fraction
  f = exp(-2 * g * m) / 2^(2 * g - 2) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * 2 * g, f) # Calculate probability
  return(pr)
}

# Function to calculate the probability of a match for once removed relatives
p_match_or = function(g, m, min_num_seg) 
{
  m = m / 100 # Convert m from cM to fraction
  f = exp(-(2 * g + 1) * m) / 2^(2 * g - 1) # Calculate f value
  pr = 1 - pbinom(min_num_seg - 1, num_chrs + genome_size * (2 * g + 1), f) # Calculate probability
  return(pr)
}

# Function to calculate the coverage of the database
coverage = function(Ks, maxg, N_pop, r, m, min_num_seg, min_num_rel, rep_direct = rep(1, 10), rep_or = rep(1, 10))
{
  N = N_pop / 2 # Convert population size to couple size
  pr_succ = length(Ks) # Initialize the vector of probabilities of success
  
  # Loop through database sizes
  for (i in 1:length(Ks)) 
  {
    K = Ks[i] # Current database size
    K_same = round(K * (r / 2) / (1 + r / 2)) # Calculate the number of direct relatives
    K_or = round(K * 1 / (1 + r / 2)) # Calculate the number of once removed relatives
    
    # Initialize vectors for probability calculations
    p_no_coal = numeric(maxg) 
    p_coal = numeric(maxg)
    p_no_coal_or = numeric(maxg) 
    p_coal_or = numeric(maxg)
    Ns = N * (r / 2)^(-(1:(maxg + 1))) # Calculate Ns values
    tot_p = 0
    tot_p_or = 0
    
    # Loop through generations
    for (g in 1:maxg)
    {
      f = 2^(2 * g - 2) / Ns[g] # Calculate f value for direct relatives
      f_or = 2^(2 * g - 1) / Ns[g + 1] # Calculate f value for once removed relatives
      
      # Update probabilities
      if (g &gt; 1) {
        p_coal[g] = p_no_coal[g - 1] * f
        p_no_coal[g] = p_no_coal[g - 1] * (1 - f)
        p_coal_or[g] = p_no_coal_or[g - 1] * f_or
        p_no_coal_or[g] = p_no_coal_or[g - 1] * (1 - f_or)
      } else {
        p_coal[g] = f 
        p_no_coal[g] = 1 - f 
        p_coal_or[g] = f_or 
        p_no_coal_or[g] = 1 - f_or
      }
      
      # Update total probabilities
      tot_p = tot_p + p_coal[g] * p_match(g, m, min_num_seg) * rep_direct[g] 
      if (g &lt; maxg) {
        tot_p_or = tot_p_or + p_coal_or[g] * p_match_or(g, m, min_num_seg) * rep_or[g]
      } 
    }

    # Calculate probability of no success
    pr_no_succ = 0
    for (n in 0:(min_num_rel - 1)) 
    {
      for (n_or in 0:n) 
      {
        pr_no_succ = pr_no_succ + dbinom(n_or, K_or, tot_p_or) * dbinom(n - n_or, K_same, tot_p)
      }
    }
    
    # Calculate the probability of success
    pr_succ[i] = 1 - pr_no_succ
  }
  
  return(pr_succ)
}

# Ks: A vector of database sizes
# maxg: Maximum relatedness to consider (1: sibs, 2: 1st cousins, 3: 2nd cousins...)
# N: Population size

# r: Mean number of children per mating pair (=per family), so 2 for a constant size population, &gt;2 for expanding population, &lt;1 for contracting population
# m: Maximum length in cM of a detectable segment
# min_num_seg: Minimum number of segments to declare a match
# min_num_rel: Minimum number of detected matches (=relatives)
# to declare success of identification</code></pre>
<pre class="r"><code>N = 250000000 #population size
N_pop = 250000000 # added by Tina to run below
num_K = 10000 #number of data points between 0 to 1 
m = 6 #minimal cM
min_num_seg = 2 #number of segments
r = 2.5 #number of kids per couple
Ks = round(seq(from=N/num_K,to=N,length.out=num_K))


c1 = coverage(Ks,maxg=2,N_pop,r,m,min_num_seg=2,min_num_rel=1) 
c2 = coverage(Ks,maxg=3,N_pop,r,m,min_num_seg=2,min_num_rel=1) 
c3 = coverage(Ks,maxg=4,N_pop,r,m,min_num_seg=2,min_num_rel=1) 
c4 = coverage(Ks,maxg=5,N_pop,r,m,min_num_seg=2,min_num_rel=1)</code></pre>
<pre class="r"><code>library(ggplot2)
# Combine the data points into a single data frame
plot_data &lt;- data.frame(Ks = rep(Ks, 4),
                        Coverage = c(c1, c2, c3, c4),
                        Relationship = factor(rep(c(&quot;1C&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;4C&quot;), each = num_K)))

# Calculate database size as a proportion of the population size
plot_data$Ks_proportion &lt;- plot_data$Ks / N

# Create a ggplot2 plot
p &lt;- ggplot(plot_data, aes(x = Ks_proportion, y = Coverage, color = Relationship, group = Relationship)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(breaks = seq(0, 0.05, 0.01), limits = c(0, 0.05)) +
  labs(title = &quot;Long-range Familial Search Coverage&quot;,
       x = &quot;Database Size / Population Size&quot;,
       y = &quot;Probability of a Match&quot;,
       color = &quot;Cousin Relationship&quot;) +
  theme_minimal() +
  theme(legend.position = &quot;none&quot;) +
  geom_text(data = subset(plot_data, Ks_proportion == 0.007), aes(label = Relationship), vjust = 3)

# Display the plot
p</code></pre>
<pre><code>Warning: Removed 38000 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 38000 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/methods.Rmd/erlich-fig1b-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-erlich-fig1b-plot-1">
Past versions of erlich-fig1b-plot-1.png
</button>
</p>
<div id="fig-erlich-fig1b-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/d97facec79dc362c30390891b60482d208807811/docs/figure/methods.Rmd/erlich-fig1b-plot-1.png" target="_blank">d97face</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Differences in the assumptions between the two models are mainly in
the following:</p>
<p>Population size: Coop and Edge’s model uses data on US population
sizes by year to estimate the number of grandparents by generation,
whereas Erlich et al assume a constant population size with a mean
number of children per couple equal to 2.5.</p>
<p>Genomic segments inheritance: Coop and Edge’s model calculates the
expected number of shared blocks of genetic material between cousins of
varying degrees of relatedness, and it uses a Poisson distribution
assumption to estimate the probability of two cousins sharing at least
one, two, or three blocks of genetic material. Erlich et al assume that
genomic segments are inherited independently of each other and models
the probability of a match using a binomial distribution.</p>
<pre class="r"><code># Compute the coverage for each degree of cousin (1st to 8th degree)
cousin_probabilities &lt;- list()
for (i in 1:8) {
  cousin_probabilities[[i]] &lt;- coverage(Ks, maxg=(i+1), N_pop, r, m, min_num_seg=2, min_num_rel=1)
}

# Create a dataframe to store the results
erlich_results &lt;- data.frame()

# Loop through each degree of cousin and add the results to the dataframe
for (i in 1:8) {
  coverage_result &lt;- cousin_probabilities[[i]]

  temp_df &lt;- data.frame(
    &quot;Degree of Cousin&quot; = rep(i, length(Ks)),
    &quot;Database Size&quot; = Ks,
    &quot;Probability&quot; = coverage_result
  )

  erlich_results &lt;- rbind(erlich_results, temp_df)
}

# Print the resulting dataframe
head(erlich_results)</code></pre>
<pre><code>  Degree.of.Cousin Database.Size Probability
1                1         25000 0.001110498
2                1         50000 0.002219763
3                1         75000 0.003327796
4                1        100000 0.004434598
5                1        125000 0.005540137
6                1        150000 0.006644482</code></pre>
<pre class="r"><code># Set up vector for cousin degree
p &lt;- c(1:8) 

# Set up initial population size
N &lt;- 76e6 

# Read in data on US population sizes by year
US_pop &lt;- read.csv(paste(path, &quot;US_popsize.csv&quot;, sep = &quot;&quot;))

# Calculate number of grandparents by generation
p_grandpar_gen &lt;- 1950 - 30 * (p + 1)

# Determine which years match the grandparents&#39; generation
these.years &lt;- match(p_grandpar_gen, US_pop$Year)

# Get population sizes by year for grandparents&#39; generation
US_Ns &lt;- US_pop$Population[these.years]

# Scale population size down by 50% (assumed number of potential parents) and 90% of those have children 
N &lt;- US_Ns
N &lt;- N * 0.5 * 0.9 

# Set minimum population size for small populations
N[US_Ns &lt; 1e6] &lt;- 1e6 # is this in the correct order? 

# Set up vector of database sizes to test
DB.sizes &lt;- c(1e6, 5e6, 10e6)
# Constants
DB.sizes &lt;- c(1000000, 5000000, 10000000, 20000000, 50000000)  # Database sizes
p &lt;- 1:8                                                      # Degree of relatedness (1st to 8th cousins)

# Calculate probabilities of having at least one p-th cousin in the database
prob_no_rellys &lt;- sapply(1:length(DB.sizes), function(i) {
  DB.size &lt;- DB.sizes[i]
  sapply(p, function(p) {
    N_p &lt;- N[p]
    1 - exp(-2^(2*p - 2) * DB.size / N_p)
  })
})

# Create a dataframe to store the results
coop_edge_results &lt;- data.frame()

# Loop through each degree of cousin and database size, and add the results to the dataframe
for (i in 1:length(p)) {
  for (j in 1:length(DB.sizes)) {
    temp_df &lt;- data.frame(
      &quot;Degree of Cousin&quot; = p[i],
      &quot;Database Size&quot; = DB.sizes[j],
      &quot;Probability&quot; = prob_no_rellys[i, j]
    )
    coop_edge_results &lt;- rbind(coop_edge_results, temp_df)
  }
}

# Print the resulting dataframe
head(coop_edge_results)</code></pre>
<pre><code>  Degree.of.Cousin Database.Size Probability
1                1         1e+06  0.03466945
2                1         5e+06  0.16173710
3                1         1e+07  0.29731531
4                1         2e+07  0.50623423
5                1         5e+07  0.82868222
6                2         1e+06  0.24625082</code></pre>
<pre class="r"><code># ... (insert code to load libraries, set up variables, read data, and calculate results) ...

# Filter coop_edge_results and erlich_results for specified database sizes
coop_edge_filtered &lt;- coop_edge_results[coop_edge_results$Database.Size %in% c(1e6, 5e6, 10e6), ]
erlich_filtered &lt;- erlich_results[erlich_results$Database.Size %in% c(1e6, 5e6, 10e6), ]

# Set up the plot layout
par(mfrow = c(1, 2))

# Plot coop_edge_results
plot(c(1, 8), c(0, 1), type = &quot;n&quot;, xlab = &quot;p (Degree of Cousin)&quot;, ylab = &quot;Probability of at least one p-th cousin in database&quot;, main = &quot;Coop &amp; Edge Results&quot;)
colors &lt;- c(&quot;blue&quot;, &quot;orange&quot;, &quot;red&quot;)
i &lt;- 1
for (db_size in c(1e6, 5e6, 10e6)) {
  temp_df &lt;- coop_edge_filtered[coop_edge_filtered$Database.Size == db_size, ]
  points(temp_df$Degree.of.Cousin, temp_df$Probability, type = &quot;b&quot;, pch = 19, lwd = 1.5, col = colors[i])
  i &lt;- i + 1
}
legend(x = &quot;bottomright&quot;, legend = c(&quot;Database size (Millions) =&quot;, &quot;1&quot;, &quot;5&quot;, &quot;10&quot;), col = c(NA, colors), pch = 19)

# Plot erlich_results
plot(c(1, 8), c(0, 1), type = &quot;n&quot;, xlab = &quot;p (Degree of Cousin)&quot;, ylab = &quot;Probability of at least one p-th cousin in database&quot;, main = &quot;Erlich Results&quot;)
i &lt;- 1
for (db_size in c(1e6, 5e6, 10e6)) {
  temp_df &lt;- erlich_filtered[erlich_filtered$Database.Size == db_size, ]
  points(temp_df$Degree.of.Cousin, temp_df$Probability, type = &quot;b&quot;, pch = 19, lwd = 1.5, col = colors[i])
  i &lt;- i + 1
}</code></pre>
<p><img src="figure/methods.Rmd/unnamed-chunk-2-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/dbad021423e088080ae2981dd213b0e26fa82c84/docs/figure/methods.Rmd/unnamed-chunk-2-1.png" target="_blank">dbad021</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Reset plotting parameters
par(mfrow = c(1, 1), mar = c(5.1, 4.1, 4.1, 2.1))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.4.4     wesanderson_0.3.7 workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] sass_0.4.8        utf8_1.2.4        generics_0.1.3    stringi_1.8.3    
 [5] digest_0.6.34     magrittr_2.0.3    evaluate_0.23     grid_4.3.2       
 [9] fastmap_1.1.1     rprojroot_2.0.4   jsonlite_1.8.8    processx_3.8.3   
[13] whisker_0.4.1     ps_1.7.5          promises_1.2.1    httr_1.4.7       
[17] fansi_1.0.6       scales_1.3.0      jquerylib_0.1.4   cli_3.6.2        
[21] rlang_1.1.3       munsell_0.5.0     withr_2.5.2       cachem_1.0.8     
[25] yaml_2.3.8        tools_4.3.2       dplyr_1.1.4       colorspace_2.1-0 
[29] httpuv_1.6.13     vctrs_0.6.5       R6_2.5.1          lifecycle_1.0.4  
[33] git2r_0.33.0      stringr_1.5.1     fs_1.6.3          pkgconfig_2.0.3  
[37] callr_3.7.3       pillar_1.9.0      bslib_0.6.1       later_1.3.2      
[41] gtable_0.3.4      glue_1.7.0        Rcpp_1.0.12       xfun_0.41        
[45] tibble_3.2.1      tidyselect_1.2.0  highr_0.10        rstudioapi_0.15.0
[49] knitr_1.45        farver_2.1.1      htmltools_0.5.7   rmarkdown_2.25   
[53] labeling_0.4.3    compiler_4.3.2    getPass_0.2-4    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
