<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Simulation of STR Pairs and Calculation of Likelihood Ratios</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="demo-data.html">Demographic Data</a>
</li>
<li>
  <a href="database-composition.html">Database composition</a>
</li>
<li>
  <a href="STR-simulation.html">STR simulations</a>
</li>
<li>
  <a href="CODIS-STR-freqs.html">CODIS STRs</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tinalasisi/PODFRIDGE">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation of STR Pairs and Calculation of
Likelihood Ratios</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2024-03-03 20:02:58</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtree2596546a55e9369c81787cb7f9f35c660272aa16targetblank2596546a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/2596546a55e9369c81787cb7f9f35c660272aa16" target="_blank">2596546</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtree2596546a55e9369c81787cb7f9f35c660272aa16targetblank2596546a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/2596546a55e9369c81787cb7f9f35c660272aa16" target="_blank">2596546</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    output/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/STR-simulation.Rmd</code>) and
HTML (<code>docs/STR-simulation.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/2596546a55e9369c81787cb7f9f35c660272aa16/analysis/STR-simulation.Rmd" target="_blank">2596546</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-03
</td>
<td>
wflow_publish("analysis/*", republish = TRUE, all = TRUE, verbose =
TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document provides a comprehensive guide on simulating Short
Tandem Repeat (STR) pairs, and calculating shared alleles, relatedness
scores, and likelihood ratios across different populations. It aims to
elucidate the methodologies applied in forensic genetics for estimating
the degree of relationship between individuals using genetic
markers.</p>
</div>
<div id="background" class="section level2">
<h2>Background</h2>
<div id="understanding-likelihood-ratios-shared-alleles-and-relatedness"
class="section level3">
<h3>Understanding Likelihood Ratios, Shared Alleles, and
Relatedness</h3>
<p>The likelihood ratio (LR) is a evaluates the probability of observing
genetic evidence under two competing hypotheses, typically pertaining to
the relationship between individuals. This ratio assists in determining
whether the genetic evidence supports a specific relationship (e.g.,
parent-child, siblings) between two individuals more than being
unrelated.</p>
<ul>
<li><strong>Shared alleles</strong> refer to alleles (versions of a
gene) that are identical by descent, meaning they are inherited from a
common ancestor.</li>
<li><strong>Relatedness scores</strong> quantify the degree of genetic
similarity between individuals, indicating potential familial
relationships.</li>
<li><strong>Likelihood ratios (LR)</strong> compare the probability of
observing the genetic evidence if two individuals have a specified
relationship versus if they are unrelated.</li>
</ul>
</div>
<div id="from-weight-of-evidence-for-forensic-dna-profiles-book"
class="section level3">
<h3>From Weight-of-evidence for forensic DNA profiles book</h3>
<p>Likelihood ratio for a single locus is:</p>
<p><span class="math display">\[
R=\kappa_0+\kappa_1 / R_X^p+\kappa_2 / R_X^u
\]</span> Where <span class="math inline">\(\kappa\)</span> is the
probability of having 0, 1 or 2 alleles IBD for a given
relationship.</p>
<p>The <span class="math inline">\(R_X\)</span> terms are quantifying
the “surprisingness” of a particular pattern of allele sharing.</p>
<p>The <span class="math inline">\(R_X^p\)</span> terms attached to the
<span class="math inline">\(kappa_1\)</span> are defined in the
following table:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\text { Table 7.2 Single-locus LRs for paternity when }
\mathcal{C}_M \text { is unavailable. }\\
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \times\left(1+2 F_{S T}\right) \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; 3 F_{S T}+\left(1-F_{S
T}\right) p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2\left(2 F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right) \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4\left(F_{S T}+\left(1-F_{S
T}\right) p_A\right)\left(F_{S T}+\left(1-F_{S T}\right) p_B\right)
/\left(2 F_{S T}+\left(1-F_{S T}\right)\left(p_A+p_B\right)\right) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>For our purposes we will take out the <span
class="math inline">\(F_{S T}\)</span> values. So the table will be as
follows:</p>
<p><span class="math display">\[
\begin{aligned}
&amp;\begin{array}{llc}
\hline c &amp; Q &amp; R_X \\
\hline \mathrm{AA} &amp; \mathrm{AA} &amp; p_A \\
\mathrm{AA} &amp; \mathrm{AB} &amp; 2 p_A \\
\mathrm{AB} &amp; \mathrm{AA} &amp; 2p_A \\
\mathrm{AB} &amp; \mathrm{AC} &amp; 4p_A \\
\mathrm{AB} &amp; \mathrm{AB} &amp; 4 p_A p_B/(p_A+p_B) \\
\hline
\end{array}
\end{aligned}
\]</span></p>
<p>If none of the alleles match, then the <span
class="math inline">\(\kappa_1 / R_X^p = 0\)</span>.</p>
<p>The <span class="math inline">\(R_X^u\)</span> terms attached to the
<span class="math inline">\(kappa_2\)</span> are defined as:</p>
<p>If both alleles match and are homozygous the equation is 6.4 (pg 85).
Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AA}\)</span>
<span class="math display">\[
\frac{\left(2 F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(3 F_{S
T}+\left(1-F_{S T}\right) p_A\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to: <span class="math display">\[
p_A{ }^2
\]</span></p>
<p>If both alleles match and are heterozygous, the equation is 6.5 (pg
85) Single locus match probability: <span
class="math inline">\(\mathrm{CSP}=\mathcal{G}_Q=\mathrm{AB}\)</span>
<span class="math display">\[
2 \frac{\left(F_{S T}+\left(1-F_{S T}\right) p_A\right)\left(F_{S
T}+\left(1-F_{S T}\right) p_B\right)}{\left(1+F_{S T}\right)\left(1+2
F_{S T}\right)}
\]</span> Simplified to:</p>
<p><span class="math display">\[
2 p_A p_B
\]</span> If both alleles do not match then <span
class="math inline">\(\kappa_2 / R_X^u = 0\)</span>.</p>
</div>
</div>
<div id="rscript-for-simulation" class="section level2">
<h2>Rscript for simulation</h2>
<p>The R script “code/known-vs-tested_simulation_script.R” simulates STR
pairs and calculates likelihood ratios, shared alleles, and relatedness
scores across different populations. The script is divided into the
following sections:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Initialization</strong>: Load required R packages and set
the working directory.</p></li>
<li><p><strong>Parameter Configuration</strong>: Input parameters for
the number of simulations for unrelated and related pairs are specified,
allowing adjustable sample sizes.</p></li>
<li><p><strong>Data Preparation</strong>: Aggregate CODIS allele
frequency data from CSV files representing different populations into a
single dataset.</p></li>
<li><p><strong>IBD Probability Definitions</strong>: Construct a table
detailing the probabilities of sharing zero, one, or two alleles
identical by descent (IBD) for various relationship types.</p></li>
<li><p><strong>Allele Frequency Calculation</strong>: Calculate allele
frequencies within the population for each simulated pair to determine
the likelihood of observing specific allele combinations.</p></li>
<li><p><strong>Likelihood Ratio Computation</strong>: Calculate
likelihood ratios for each pair using allele frequencies and IBD
probabilities, focusing on <span class="math inline">\(R_{Xp}\)</span>
and <span class="math inline">\(R_{Xu}\)</span> values.</p></li>
<li><p><strong>Simulation of Individual Pairs</strong>: Generate alleles
for individuals based on population-specific allele frequencies to
simulate STR pairs.</p></li>
<li><p><strong>Shared Allele Analysis</strong>: Evaluate the number of
alleles shared between pairs, categorizing them by their genetic
relationship.</p></li>
<li><p><strong>Aggregate Analysis and Visualization</strong>: Summarize
the results and generate visual representations, such as box plots and
heatmaps, to illustrate the relationship between known and tested
relationships across populations.</p></li>
<li><p><strong>Result Exportation</strong>: Export aggregated results as
CSV files for further analysis.</p></li>
</ol>
<p>Due to the computational cost of simulating large numbers of STR
pairs, the script has been run separately and we visualize the results
below.</p>
</div>
<div id="simulation-results" class="section level2">
<h2>Simulation results</h2>
<pre><code>Rows: 360000 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): population, known_relationship_type, tested_relationship_type
dbl (3): replicate_id, num_shared_alleles_sum, log_R_sum

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div
id="proportion-of-individuals-of-known-relationship-type-exceeding-likelihood-cut-off"
class="section level2">
<h2>Proportion of individuals of known relationship type exceeding
likelihood cut-off</h2>
<pre><code>  population relationship_type fp_rate prop_exceeding
1       AfAm      parent_child     0.1          1.000
2      Asian      parent_child     0.1          1.000
3       Cauc      parent_child     0.1          1.000
4   Hispanic      parent_child     0.1          1.000
5       AfAm     full_siblings     0.1          0.794
6      Asian     full_siblings     0.1          0.820</code></pre>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cut-offs-for-each-fpr" class="section level2">
<h2>Cut-offs for each FPR</h2>
<pre><code>`summarise()` has grouped output by &#39;population&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 24 × 7
   population known_relationship_type mean_log_R_sum median_log_R_sum
   &lt;chr&gt;      &lt;chr&gt;                            &lt;dbl&gt;            &lt;dbl&gt;
 1 AfAm       cousins                         -0.533            2.42 
 2 AfAm       full_siblings                   10.5             11.5  
 3 AfAm       half_siblings                    3.85             6.80 
 4 AfAm       parent_child                    19.6             19.0  
 5 AfAm       second_cousins                  -1.23             1.78 
 6 AfAm       unrelated                       -2.00             0.914
 7 Asian      cousins                         -0.538            2.42 
 8 Asian      full_siblings                   10.5             11.6  
 9 Asian      half_siblings                    3.82             6.78 
10 Asian      parent_child                    19.5             18.9  
# ℹ 14 more rows
# ℹ 3 more variables: min_log_R_sum &lt;dbl&gt;, max_log_R_sum &lt;dbl&gt;, count &lt;int&gt;</code></pre>
<div
id="calculate-llr-cutoffs-for-unrelated-pairs-at-each-fpr-by-population"
class="section level4">
<h4>Calculate LLR cutoffs for unrelated pairs at each FPR by
population</h4>
<p>This table presents aggregated log_R_sum values by relationship type
for each population, alongside LLR cutoffs calculated for unrelated
pairs to maintain false positive rates (FPRs) of 1%, 0.1%, and 0.01%.
The cutoffs indicate the LLR threshold above which a pair is less likely
to be unrelated at the given FPR, thereby serving as a critical
benchmark for assessing relationship evidence. The mean log_R_sum values
further elucidate the average strength of genetic evidence supporting
each relationship type within populations, highlighting variations and
consistencies in genetic relatedness indicators across demographic
groups.</p>
<pre><code>
Attaching package: &#39;kableExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
population
</th>
<th style="text-align:right;">
cousins
</th>
<th style="text-align:right;">
full_siblings
</th>
<th style="text-align:right;">
half_siblings
</th>
<th style="text-align:right;">
parent_child
</th>
<th style="text-align:right;">
second_cousins
</th>
<th style="text-align:right;">
unrelated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
AfAm
</td>
<td style="text-align:right;">
-0.5334790
</td>
<td style="text-align:right;">
10.45417
</td>
<td style="text-align:right;">
3.849582
</td>
<td style="text-align:right;">
19.62211
</td>
<td style="text-align:right;">
-1.229432
</td>
<td style="text-align:right;">
-2.003701
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Asian
</td>
<td style="text-align:right;">
-0.5379709
</td>
<td style="text-align:right;">
10.45853
</td>
<td style="text-align:right;">
3.821000
</td>
<td style="text-align:right;">
19.54267
</td>
<td style="text-align:right;">
-1.387000
</td>
<td style="text-align:right;">
-1.982983
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Cauc
</td>
<td style="text-align:right;">
-0.5075911
</td>
<td style="text-align:right;">
10.48275
</td>
<td style="text-align:right;">
3.873924
</td>
<td style="text-align:right;">
19.51039
</td>
<td style="text-align:right;">
-1.218428
</td>
<td style="text-align:right;">
-1.982207
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Hispanic
</td>
<td style="text-align:right;">
-0.5536143
</td>
<td style="text-align:right;">
10.40972
</td>
<td style="text-align:right;">
3.837237
</td>
<td style="text-align:right;">
19.58039
</td>
<td style="text-align:right;">
-1.249979
</td>
<td style="text-align:right;">
-1.984905
</td>
</tr>
</tbody>
</table>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
<p><img src="figure/STR-simulation.Rmd/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.4.0   patchwork_1.2.0    lubridate_1.9.3    forcats_1.0.0     
 [5] stringr_1.5.1      dplyr_1.1.4        purrr_1.0.2        readr_2.1.5       
 [9] tidyr_1.3.0        tibble_3.2.1       ggplot2_3.4.4      tidyverse_2.0.0   
[13] RColorBrewer_1.1-3 wesanderson_0.3.7  workflowr_1.7.1   

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.41         bslib_0.6.1       processx_3.8.3   
 [5] callr_3.7.3       tzdb_0.4.0        vctrs_0.6.5       tools_4.3.2      
 [9] ps_1.7.5          generics_0.1.3    parallel_4.3.2    fansi_1.0.6      
[13] highr_0.10        pkgconfig_2.0.3   lifecycle_1.0.4   compiler_4.3.2   
[17] farver_2.1.1      git2r_0.33.0      munsell_0.5.0     getPass_0.2-4    
[21] httpuv_1.6.13     htmltools_0.5.7   sass_0.4.8        yaml_2.3.8       
[25] later_1.3.2       pillar_1.9.0      crayon_1.5.2      jquerylib_0.1.4  
[29] whisker_0.4.1     cachem_1.0.8      tidyselect_1.2.0  digest_0.6.34    
[33] stringi_1.8.3     labeling_0.4.3    rprojroot_2.0.4   fastmap_1.1.1    
[37] grid_4.3.2        colorspace_2.1-0  cli_3.6.2         magrittr_2.0.3   
[41] utf8_1.2.4        withr_2.5.2       scales_1.3.0      promises_1.2.1   
[45] bit64_4.0.5       timechange_0.2.0  rmarkdown_2.25    httr_1.4.7       
[49] bit_4.0.5         hms_1.1.3         evaluate_0.23     knitr_1.45       
[53] viridisLite_0.4.2 rlang_1.1.3       Rcpp_1.0.12       glue_1.7.0       
[57] xml2_1.3.6        svglite_2.1.3     rstudioapi_0.15.0 vroom_1.6.5      
[61] jsonlite_1.8.8    R6_2.5.1          systemfonts_1.0.5 fs_1.6.3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
