<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Short Range Familial Search</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PODFRIDGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="demo-data.html">Demographic Data</a>
</li>
<li>
  <a href="analyses.html">Analyses</a>
</li>
<li>
  <a href="database-composition.html">Database composition</a>
</li>
<li>
  <a href="shortrange.html">Short Range</a>
</li>
<li>
  <a href="manuscript_figures.html">Manuscript Figures</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://tinalasisi.github.io/PODFRIDGE/">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Short Range Familial Search</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2024-03-02 19:40:40</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PODFRIDGE/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230302code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230302)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230302code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230302)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtreeaa3ff5cd58cd98d8f1274294dfd38b244f01c65btargetblankaa3ff5ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/aa3ff5cd58cd98d8f1274294dfd38b244f01c65b" target="_blank">aa3ff5c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiPODFRIDGEtreeaa3ff5cd58cd98d8f1274294dfd38b244f01c65btargetblankaa3ff5ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tinalasisi/PODFRIDGE/tree/aa3ff5cd58cd98d8f1274294dfd38b244f01c65b" target="_blank">aa3ff5c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    output/.DS_Store

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/demo-data.Rmd
    Modified:   data/us-pop-detailed-multisheet.xls

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/STR.Rmd</code>) and HTML
(<code>docs/STR.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/aa3ff5cd58cd98d8f1274294dfd38b244f01c65b/docs/STR.html" target="_blank">aa3ff5c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/e4c698e5b7d0702fee532f0800fb7bc9daeacf1c/analysis/STR.Rmd" target="_blank">e4c698e</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-02-27
</td>
<td>
Publish new pages + update plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/PODFRIDGE/e4c698e5b7d0702fee532f0800fb7bc9daeacf1c/docs/STR.html" target="_blank">e4c698e</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-02-27
</td>
<td>
Publish new pages + update plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/acfb421286f581facbf8c7e0c912f6d8bc66f628/analysis/STR.Rmd" target="_blank">acfb421</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-06-06
</td>
<td>
Update simulation script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/d959019be3a6eb5dd858286a593c0cae129a7400/analysis/STR.Rmd" target="_blank">d959019</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-05-31
</td>
<td>
Add STR.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="codis-marker-allele-frequencies" class="section level2">
<h2>CODIS marker allele frequencies</h2>
<p>Frequencies and raw genotypes for different populations were found <a
href="https://www.promega.com/products/pm/genetic-identity/population-statistics/allele-frequencies/">here</a>
and refer to Steffen, C.R., Coble, M.D., Gettings, K.B., Vallone, P.M.
(2017) Corrigendum to ‘U.S. Population Data for 29 Autosomal STR Loci’
[Forensic Sci. Int. Genet. 7 (2013) e82-e83]. Forensic Sci. Int. Genet.
31, e36–e40. The US core CODIS markers are a subset of the 29 described
here.</p>
</div>
<div id="load-codis-allele-frequencies" class="section level2">
<h2>Load CODIS allele frequencies</h2>
<p>CODIS allele frequencies were found through NIST STR base and
specifically downloaded from the supplementary materials of Steffen et
al 2017. These are 1036 unrelated individuals from the U.S.
population.</p>
<pre class="r"><code># Define the file paths
file_paths &lt;- list.files(path = &quot;data&quot;, pattern = &quot;1036_.*\\.csv&quot;, full.names = TRUE)

# Create a list of data frames
df_list &lt;- lapply(file_paths, function(path) {
  read_csv(path, col_types = cols(
    marker = col_character(),
    allele = col_double(),
    frequency = col_double(),
    population = col_character()
  ))
})

# Bind all data frames into one
df &lt;- bind_rows(df_list)

df_freq &lt;- df</code></pre>
</div>
<div id="simulating-genotypes" class="section level2">
<h2>Simulating genotypes</h2>
<pre class="r"><code>df_ibdprobs &lt;- tibble(
  relationship = 
    c(&quot;parent_child&quot;, &quot;full_siblings&quot;, &quot;half_siblings&quot;, &quot;cousins&quot;, &quot;second_cousins&quot;, &quot;unrelated&quot;),
  k0 = c(0, 1/4, 1/2, 7/8, 15/16, 1),
  k1 = c(1, 1/2, 1/2, 1/8, 1/16, 0),
  k2 = c(0, 1/4, 0, 0, 0, 0)
)

df_ibdprobs</code></pre>
<pre><code># A tibble: 6 × 4
  relationship      k0     k1    k2
  &lt;chr&gt;          &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 parent_child   0     1       0   
2 full_siblings  0.25  0.5     0.25
3 half_siblings  0.5   0.5     0   
4 cousins        0.875 0.125   0   
5 second_cousins 0.938 0.0625  0   
6 unrelated      1     0       0   </code></pre>
<pre class="r"><code>simulate_STRpairs &lt;- function(population, relationship_type, df_allelefreq, df_ibdprobs, n_sims=1) {
  
  markers &lt;- unique(df_allelefreq$marker)
  
  allele_frequencies_by_marker &lt;- df_allelefreq %&gt;%
    filter(population == population) %&gt;%
    split(.$marker)

  allele_frequencies_by_marker &lt;- map(allele_frequencies_by_marker, ~.x %&gt;% pull(frequency) %&gt;% setNames(.x$allele))

  prob_shared_alleles &lt;- df_ibdprobs %&gt;%
    filter(relationship == relationship) %&gt;%
    select(k0, k1, k2) %&gt;%
    unlist() %&gt;%
    as.numeric()
  non_zero_indices &lt;- which(prob_shared_alleles != 0)

  k0 &lt;- prob_shared_alleles[1]
  k1 &lt;- prob_shared_alleles[2]
  k2 &lt;- prob_shared_alleles[3]

  results &lt;- lapply(markers, function(current_marker) {
    marker_results &lt;- lapply(seq_len(n_sims), function(replicate_id) {
      individual1 &lt;- setNames(
        lapply(markers, function(current_marker) {
          allele_frequencies &lt;- allele_frequencies_by_marker[[current_marker]]
          return(sample(names(allele_frequencies), size = 2, replace = TRUE, prob = allele_frequencies))
        }), markers)
      
      individual2 &lt;- setNames(
        lapply(markers, function(current_marker) {
          allele_frequencies &lt;- allele_frequencies_by_marker[[current_marker]]
          num_shared_alleles &lt;- sample(non_zero_indices - 1, size = 1, prob = prob_shared_alleles[non_zero_indices])
          alleles_from_individual1 &lt;- sample(individual1[[current_marker]], size = num_shared_alleles)
          alleles_from_population &lt;- sample(names(allele_frequencies), size = 2 - num_shared_alleles, replace = TRUE, prob = allele_frequencies)
          return(c(alleles_from_individual1, alleles_from_population))
        }), markers)

      ind1_alleles &lt;- individual1[[current_marker]]
      ind2_alleles &lt;- individual2[[current_marker]]
      shared_alleles &lt;- intersect(ind1_alleles, ind2_alleles)
      num_shared_alleles &lt;- length(shared_alleles)
      
      R_Xp &lt;- sum(purrr::map_dbl(shared_alleles, function(x) unlist(allele_frequencies_by_marker[[current_marker]][x])))
R_Xu &lt;- sum(purrr::map_dbl(ind1_alleles, function(x) unlist(allele_frequencies_by_marker[[current_marker]][x])) * purrr::map_dbl(ind2_alleles, function(x) unlist(allele_frequencies_by_marker[[current_marker]][x])))

      R &lt;- k0
      if (R_Xp != 0) { R &lt;- R + (k1 / R_Xp) }
      if (R_Xu != 0) { R &lt;- R + (k2 / R_Xu) }
      
      log_R &lt;- log(R)
      
       # Add the replicate_id column to the output tibble
      return(tibble(population = population,
                relationship_type = relationship_type,
                marker = current_marker,
                num_shared_alleles = num_shared_alleles,
                log_R = log_R,
                replicate_id = replicate_id)) # Add this line
})
    
    marker_results &lt;- bind_rows(marker_results)
    return(marker_results)
  })

  result &lt;- bind_rows(results)

  # Aggregate results for each replicate, summing num_shared_alleles and log_R values.
  result_by_replicate &lt;- result %&gt;%
    group_by(population, relationship_type, replicate_id) %&gt;%
    summarise(num_shared_alleles_sum = sum(num_shared_alleles),
              log_R_sum = sum(log_R),
              .groups = &quot;drop&quot;)
  
  return(result_by_replicate)

}</code></pre>
</div>
<div id="combinations" class="section level2">
<h2>Combinations</h2>
<pre class="r"><code>simulation_combinations &lt;- function(df, n_sims_unrelated, n_sims_related) {
  # Define the list of relationship types
  relationship_types &lt;- c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)
  
  # Get unique populations from the input dataframe
  unique_populations &lt;- unique(df$population)
  filtered_populations &lt;- unique_populations[unique_populations != &quot;all&quot;]
  
  # Create a dataframe of all combinations of populations and relationship types
  combinations &lt;- expand_grid(population = filtered_populations, relationship_type = relationship_types)

  # Add the number of simulations for unrelated or related relationships
  combinations$n_sims &lt;- ifelse(combinations$relationship_type == &quot;unrelated&quot;, n_sims_unrelated, n_sims_related)
  
  return(combinations)
}</code></pre>
<pre class="r"><code># Example usage
result_combinations &lt;- simulation_combinations(df, n_sims_unrelated = 10, n_sims_related = 5)

result_combinations</code></pre>
<pre><code># A tibble: 24 × 3
   population relationship_type n_sims
   &lt;chr&gt;      &lt;chr&gt;              &lt;dbl&gt;
 1 AfAm       parent_child           5
 2 AfAm       full_siblings          5
 3 AfAm       half_siblings          5
 4 AfAm       cousins                5
 5 AfAm       second_cousins         5
 6 AfAm       unrelated             10
 7 Asian      parent_child           5
 8 Asian      full_siblings          5
 9 Asian      half_siblings          5
10 Asian      cousins                5
# ℹ 14 more rows</code></pre>
</div>
<div id="parallel" class="section level2">
<h2>Parallel</h2>
<pre class="r"><code># # Apply the simulate_STRpairs function to each row in result_combinations
# results &lt;- result_combinations %&gt;%
#   pmap_dfr(function(population, relationship_type, n_sims) {
#     simulate_STRpairs(population, relationship_type, df_allelefreq = df_freq, df_ibdprobs = df_ibdprobs, n_sims = n_sims)
#   })</code></pre>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<pre class="r"><code># Function to capitalize the first letter of a string
ucfirst &lt;- function(s) {
  paste(toupper(substring(s, 1,1)), substring(s, 2), sep = &quot;&quot;)
}

create_plot &lt;- function(df, variable_to_plot, relationship_col, population_col) {
  # Set the population_shape variable as factor
  df$population_shape &lt;- factor(df[[population_col]])

  # Create the plot
  p &lt;- ggplot(df, aes(x = .data[[relationship_col]], y = .data[[variable_to_plot]], 
                      color = .data[[population_col]], shape = .data[[population_col]], fill = .data[[population_col]])) +
    geom_boxplot(alpha = 0.4, position = position_dodge(width = 0.75)) +
    geom_point(position = position_jitterdodge(jitter.width = 0.1, dodge.width = 0.75), size = 1, alpha = 0.6) +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    scale_x_discrete(limits = c(&#39;parent_child&#39;, &#39;full_siblings&#39;, &#39;half_siblings&#39;, &#39;cousins&#39;, &#39;second_cousins&#39;, &#39;unrelated&#39;)) +
    scale_shape_manual(values = c(16, 17, 15, 18)) + # Change these values to desired shapes
    labs(title = paste(ucfirst(variable_to_plot), &quot;by Relationship Type and Population&quot;),
         x = &quot;Relationship Type&quot;, 
         y = ucfirst(variable_to_plot), 
         color = &quot;Population&quot;,
         shape = &quot;Population&quot;,
         fill = &quot;Population&quot;)
  
  # Save the plot to the /output folder with a custom file name
  save_plot &lt;- function(plot, plot_name) {
    ggsave(filename = paste0(&quot;output/&quot;, plot_name, &quot;.png&quot;), plot = plot, height = 6, width = 8, units = &quot;in&quot;)
  }
  
  # Call the save_plot function to save the plot
  save_plot(p, paste(&quot;plot_&quot;, variable_to_plot, sep = &quot;&quot;))
  
  return(p)
}</code></pre>
<pre class="r"><code>simulation_results &lt;- read.csv(&quot;data/simulation_results.csv&quot;)</code></pre>
<pre class="r"><code># Filter your data for unique STR markers and remove the &quot;all&quot; population
# df_plt_final &lt;- results_parallel %&gt;%
df_plt_final &lt;- simulation_results %&gt;%
  select(-replicate_id)

p &lt;- create_plot(df_plt_final, &quot;log_R_sum&quot;, &quot;relationship_type&quot;, &quot;population&quot;)
p</code></pre>
<p><img src="figure/STR.Rmd/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/e4c698e5b7d0702fee532f0800fb7bc9daeacf1c/docs/figure/STR.Rmd/unnamed-chunk-8-1.png" target="_blank">e4c698e</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plt_allele &lt;- create_plot(df_plt_final, &quot;num_shared_alleles_sum&quot;, &quot;relationship_type&quot;, &quot;population&quot;)
plt_allele</code></pre>
<p><img src="figure/STR.Rmd/unnamed-chunk-8-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/PODFRIDGE/blob/e4c698e5b7d0702fee532f0800fb7bc9daeacf1c/docs/figure/STR.Rmd/unnamed-chunk-8-2.png" target="_blank">e4c698e</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Detroit
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] progressr_0.14.0 furrr_0.3.1      future_1.33.1    lubridate_1.9.3 
 [5] forcats_1.0.0    stringr_1.5.1    dplyr_1.1.4      purrr_1.0.2     
 [9] readr_2.1.5      tidyr_1.3.0      tibble_3.2.1     ggplot2_3.4.4   
[13] tidyverse_2.0.0  readxl_1.4.3    

loaded via a namespace (and not attached):
 [1] gtable_0.3.4      xfun_0.41         bslib_0.6.1       tzdb_0.4.0       
 [5] vctrs_0.6.5       tools_4.3.2       generics_0.1.3    parallel_4.3.2   
 [9] fansi_1.0.6       highr_0.10        pkgconfig_2.0.3   lifecycle_1.0.4  
[13] farver_2.1.1      compiler_4.3.2    git2r_0.33.0      textshaping_0.3.7
[17] munsell_0.5.0     codetools_0.2-19  httpuv_1.6.13     htmltools_0.5.7  
[21] sass_0.4.8        yaml_2.3.8        later_1.3.2       pillar_1.9.0     
[25] crayon_1.5.2      jquerylib_0.1.4   whisker_0.4.1     cachem_1.0.8     
[29] parallelly_1.36.0 tidyselect_1.2.0  digest_0.6.34     stringi_1.8.3    
[33] listenv_0.9.0     labeling_0.4.3    rprojroot_2.0.4   fastmap_1.1.1    
[37] grid_4.3.2        colorspace_2.1-0  cli_3.6.2         magrittr_2.0.3   
[41] utf8_1.2.4        withr_2.5.2       scales_1.3.0      promises_1.2.1   
[45] bit64_4.0.5       timechange_0.2.0  rmarkdown_2.25    globals_0.16.2   
[49] bit_4.0.5         workflowr_1.7.1   cellranger_1.1.0  ragg_1.2.7       
[53] hms_1.1.3         evaluate_0.23     knitr_1.45        rlang_1.1.3      
[57] Rcpp_1.0.12       glue_1.7.0        rstudioapi_0.15.0 vroom_1.6.5      
[61] jsonlite_1.8.8    R6_2.5.1          systemfonts_1.0.5 fs_1.6.3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
