---
title: "PODFRIDGE"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

The use of genetic data to identify individuals has become increasingly prevalent with the rise of consumer genomics databases and forensic genetic genealogy. Recent studies have shown that long-range familial searches can implicate the majority of US individuals of European descent (see [Erlich et al 2018](https://www.science.org/doi/10.1126/science.aau4832) and [Coop & Edge 2019](https://www.biorxiv.org/content/10.1101/531384v1)). These findings have motivated us to carry out further investigations into the disparities between populations in the likelihood of identifying individuals through genetic relatives. 

African-Americans are overrepresented in forensic databases (see [Murphy and Tong, 2019](https://californialawreview.org/print/racial-composition-forensic-dna-databases/)
), but what we know about direct-to-consumer databases suggests that Europeans are highly represented there. 

Our study aims to examine the impact of the varying degrees of representation across different types of genetic databases on the potential consequences for different populations, as well as the impact of different types of genetic data such as SNP genotyping and STR markers.

The PODFRIDGE (**Po**pulation **D**ifferences in **F**orensic **R**elative **Id**entification via **Ge**netic Genealogy) project aims to address this question by using mathematical estimations of genetic relatedness based on previous methods, as well as simulating pedigrees and genomes for two populations based on input parameters. Additionally, demographic data from the US Census microdata will be used to obtain population size information on European and African Americans.

Our study aims to provide insight into the implications of these techniques for different populations and contribute to the ongoing discussions around genetic privacy and ethics.


### Database Representation

The reanalysis of the data from Murphy and Tong (2019) can be found at [Murphy & Tong FOIA Data](murphy.html).

The meta-analysis on [DTC Database Composition](database-composition.html) is found in the linked page.

And our [Estimation of Forensic Database Representation](racial_proportion.html). 

### Family Structure
We analyze 4 Census years in which mothers report the total number of children born to them to estimate population variation in family size. The results are found in [Family Size Variation](relative-distribution.html).

We ask whether the difference in fertility assumptions significantly impacts relative counts in the [Fertility Assumptions](fertility_assumptions.html) and provide a [Family Size Simulator](family_size_findability.html) that can be used to simulate family sizes based on the distribution of family sizes in the US Census data.

### STR Likelihood Ratio Effects 
We visualize previously reported [CODIS STR Allele Frequencies](CODIS-STR-freqs.html) and provide a [Likelihood Ratio Calculation](lr.html) deriving the match probability equation used in forensic genetics. Our [Simulation Framework for STR loci](STR-simulation.html) is also available and our [Likelihood Ration Distributions](lr_dist.html) are also shared.

### Genetic Surveillance Burden
We review [Long-range Familial Searching Reach Estimation Methods](methods.html) used in Erlich et al 2018 and Coop & Edge 2019 and provide our own analysis in [Disparities in Long-Range Familial Search](probability_disparity.html). 

For short-range familial searchwes, we provide a [Population Estimation of Genetic Surveillance Equation](final_equation.html) that will serve as the basis for our calculations of disparities in genetic surveillance burden based on database representation and family structure.
